<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <title>RDML-Tools Help | GEAR</title>
  <meta name="description" content="Tools to handle RDML files.">
  <link rel="stylesheet" href="static/libs/bootstrap/4.0.0/css/bootstrap.min.css" crossorigin="anonymous">
  <link rel="stylesheet" href="static/libs/fontawesome-free-6.2.1-web/css/all.min.css">
  <link rel="stylesheet" href="static/css/gear.css">
</head>

<body>

  <header class="px-0 py-4">
    <div class="container">
      <h1 class="display-4 app-name">RDML-Tools Help</h1>
        <p class="lead">Information about handling RDML-Tools.</p>
        <a class="plain" href="help.html" target="_blank">
          <i class="fas fa-question"></i> Get help
        </a>
        ·
        <a class="plain" href="index.html" target="_blank">
          <i class="fas fa-landmark"></i> RDML-Tools
        </a>
        ·
        <a class="plain" href="https://www.gear-genomics.com/citation" target="_blank">
          <i class="fas fa-user-graduate"></i> Citation
        </a>
        ·
        <a class="plain" target="_blank" href="https://github.com/RDML-consortium/rdml-tools">
          <i class="fab fa-github-alt"></i> Source
        </a>
    </div>
  </header>

  <main class="container" id="helpAllContainer">
    <h4 id="Contact">Contact</h4>
    <p>
      <a class="plain" href="https://www.gear-genomics.com/contact" target="_blank">
        Contact the developers.
      </a>
    </p>
    <h4 id="Index">Table of Contents</h4>
    <ul class="toc_list">
      <li><a href="#Introduction">1. Introduction</a>
        <ul>
          <li><a href="#General-Introduction">1.1. General Introduction</a></li>
          <li><a href="#Data-Standard">1.2. RDML Data Standard</a></li>
          <li><a href="#License">1.3. License</a></li>
          <li><a href="#Citation">1.4. How to Cite</a></li>
          <li><a href="#LRP-Library">1.5. The Python Library rdmlpython and the Command Line</a></li>
          <li><a href="#Acknowledgements">1.6. Acknowledgement</a></li>
        </ul>
      </li>
      <li><a href="#Basic-RDML-Tools">2. RDML-Tools</a>
        <li><a href="#Tool-Introduction">2.1. Introduction</a>
        <ul>
          <li><a href="#Import">2.2. Import Data into RDML</a>
          <ul>
            <li><a href="#Import-Introduction">2.2.1. Introduction into Importing Data</a></li>
            <li><a href="#Import-validRDMLfile">2.2.2. Import of a Valid RDML File</a></li>
            <li><a href="#Import-invalidRDMLfile">2.2.3. Import of a Invalid RDML File</a></li>
            <li><a href="#Import-validRDESfile">2.2.4. Import of a Valid RDES File</a></li>
            <li><a href="#Import-RDES-Introduction">2.2.5. Introduction Into RDES</a></li>
            <li><a href="#Import-Raw-Data-Warning">2.2.6. A Note on Raw Data</a></li>
            <li><a href="#Import-repairRDESfile">2.2.7. Repair a RDES File</a></li>
            <li><a href="#Import-knownCSVfile">2.2.8. Import of a Known CSV File</a></li>
            <li><a href="#Import-unknownCSVfile">2.2.9. Import of a Unknown CSV File</a></li>
          </ul></li>
        <li><a href="#RDML-Edit">2.3. Supplement Remaining Data using RDML-Edit</a></li>
        <li><a href="#RDML-RunEdit">2.4. Annotating a Run in RDML-RunEdit</a></li>
        <li><a href="#RDML-Merge">2.5. Copy Annotation Using RDML-Merge</a></li>
        <li><a href="#RDML-AnnotationEdit">2.6. Annotate Experiments Using RDML-AnnotationEdit</a></li>
        <li><a href="#RDML-RunAnalysis">2.7. Analyze a Run with RDML-RunAnalysis</a>
          <ul>
            <li><a href="#RDML-RunAnalysis-Melting">2.7.1. Melting Curve Analysis with RDML-RunAnalysis</a></li>
            <li><a href="#RDML-RunAnalysis-Amplification">2.7.2. Amplification Curve Analysis with RDML-RunAnalysis</a></li>
          </ul></li>
        <li><a href="#RDML-AnnotationEdit">2.6. Annotate Experiments Using RDML-AnnotationEdit</a></li>
        <li><a href="#RDML-AnnotationEdit">2.6. Annotate Experiments Using RDML-AnnotationEdit</a></li>
        </ul>
      </li>
      <li><a href="#RDML-RunAnalysis">3. RDML-RunAnalysis</a>
        <ul>
          <li><a href="#LRP-Introduction">3.1. Introduction</a></li>
          <li><a href="#LRP-Warning">3.2. A Word of Caution</a></li>
          <li><a href="#LRP-References">3.3. LinRegPCR References</a></li>
          <li><a href="#LRP-Terms">3.4. Definitions of Terms Used in This Manual</a></li>
          <li><a href="#LRP-Theory-Behind">3.5. Theory Behind the LinRegPCR Program</a>
            <ul>
             <li><a href="#LRP-Basics">3.5.1. Basic Principle</a></li>
             <li><a href="#LRP-Monitoring-Chemistries">3.5.2. Cq and Efficiency with Different Monitoring Chemistries</a></li>
            </ul>
          </li>
          <li><a href="#LRP-Baseline">3.6. Baseline Estimation and Error Flagging</a>
            <ul>
              <li><a href="#LRP-Baseline-Basics">3.6.1 Baseline Estimation and Flagging of Deviating Reactions</a></li>
              <li><a href="#LRP-Error-Flagging">3.6.2. Error Flagging</a></li>
              <li><a href="#LRP-Failing-Baseline">3.6.3. Reasons for Failing Baseline Estimation and noisy samples</a></li>
              <li><a href="#LRP-Neg-Val-Warning">3.6.4. Negative Values Warning</a></li>
            </ul>
          </li>
          <li><a href="#LRP-Mean-Efficiency">3.7. Mean Efficiency Calculation</a></li>
          <li><a href="#LRP-WoL-Threshold">3.8. Window-of-Linearity and Quantification Threshold</a></li>
          <li><a href="#LRP-Diff-Pascal">3.9. Differences Between Pascal and Python Version</a></li>
          <li><a href="#LRP-Prepare-RDML">3.10. Importance of the information in RDML files</a></li>
          <li><a href="#LRP-Run">3.11. Run RunAnalysis</a></li>
        </ul>
      </li>
      <li><a href="#RDML-ExperimentAnalysis">4. RDML-ExperimentAnalysis</a>
        <ul>
          <li><a href="#ANA-Introduction">4.1. Introduction</a></li>
          <li><a href="#ANA-Plate-Correct">4.2. InterRunCorrection</a></li>
          <li><a href="#ANA-Absolute-Quan">4.3. AbsoluteQuantification</a></li>
        </ul>
      </li>
      <li><a href="#RDML-qPCR-Basics">5. qPCR Basics</a>
        <ul>
          <li><a href="#BASE-Introduction">5.1. Introduction</a></li>
          <li><a href="#BASE-qPCR-Setup">5.2. Setting up a qPCR</a></li>
          <li><a href="#BASE-Genomic-Std">5.3. Use genomic DNA as Standard</a></li>
        </ul>
      </li>
    </ul>

    <h4 id="Introduction">1. Introduction</h4>
    <h5 id="General-Introduction">1.1. General Introduction</h5>
    <p style="text-align: justify">
      The RDML-Tools are a software package with everything needed to analyze quantitative PCR (qPCR) data. It can import the raw data 
      exported by the qPCR machines, repair RDML-files, complement missing data, analyze melting and amplification 
      curves, combine several plates, identify reference genes, includes absolute and relative quantification and 
      evaluate experimental groups. It uses <a href="https://rdml.org" target="_blank">Real-time PCR Data Markup Language (RDML)</a>
      to store the data.
      <br />
      <br />
      RDML is a structured and universal data standard for exchanging qPCR data and digital PCR (dPCR) data. The data standard
      should contain sufficient information to understand the experimental setup, re-analyse the data and interpret the
      results. It was designed to enable transparent exchange of annotated qPCR data between instrument software and
      third-party data analysis packages, between colleagues and collaborators, and between authors, peer reviewers,
      journals and readers.
      <br />
      <br />
      The <a href="https://github.com/RDML-consortium/rdml-tools" target="_blank">RDML-Tools</a> center around the
      <a href="https://github.com/RDML-consortium/rdml-schema" target="_blank">RDML data format</a>. The
      <a href="https://github.com/RDML-consortium/rdml-tools" target="_blank">RDML-Tools</a> mainly provide a web page
      access to the functions of the
      <a href="https://github.com/RDML-consortium/rdmlpython" target="_blank">RDML-Python library</a>, designed to
      create and handle RDML files and their dependencies. The
      <a href="https://github.com/RDML-consortium/rdmlpython" target="_blank">RDML-Python library</a> also offers
      advanced analysis programs like <a href="#RDML-RunAnalysis">RunAnalysis</a> and MeltCurveAnalysis MeltCurveAnalysis
      to analyze the fluorescence data in the RDML files and provide biological or clinical results.
      <br />
      <br />
      Users may choose to install the library as a
      <a href="https://github.com/RDML-consortium/rdmlpython-package" target="_blank">python package</a> calling
      "python -m pip install rdmlpython". All software is GPL licensed except the RDML-Python library being MIT
      licensed.
      <br />
      <br />
      Andreas Untergasser (Zentrum f&uuml;r Molekulare Biologie, University of Heidelberg, Germany; Genome Core,
      EMBL, Heidelberg, Germany), Maurice van den Hoff and Jan Ruijter (both Department of Medical Biology,
      Amsterdam UMC, location, AMC, Amsterdam).
    </p>

    <h5 id="Data-Standard">1.2. RDML and RDES Data Standard</h5>
    <p style="text-align: justify">
      The <a target="_blank" href="https://rdml.org/">RDML data standard</a> is a compressed
      text file in Extensible Markup Language (XML). By design, it collects the information of different experimenters,
      dyes, samples, targets and thermal cycling conditions separately to experiment/run information, were this
      information is only referenced by the unique ID. In a final RDML file, for example, one target might be
      referenced many times. The RDML-Tools and RDML-Python library were created to handle these dependencies and
      allow modification of an ID without having to update all the occurrences in the experiment.
      <br />
      <br />
      RDML evolved over the years and there are several versions available today. The RDML-Tools can convert between
      the versions. Upload the file to RDML-Edit, activate the edit mode on the main tab and the migration buttons
      become available on the "more" tab. Version 1.0 is automatically converted to version 1.1, the lowest supported
      version by the RDML-Tools. Newer versions offer new and may remove obsolete functionality. Be aware that
      digital PCR is only supported from RDML version 1.3 onwards.
      <br />
      <br />
      RDML was first published in
      "<a href="https://www.ncbi.nlm.nih.gov/pubmed/19223324" target="_blank">RDML: structured language and
      reporting guidelines for real-time quantitative PCR data in 2009 (Lefever et al., 2009 Nucleic Acids Res.
      37(7):2065-2069. (doi: 10.1093/nar/gkp056 or PMID: 19223324))</a>".
      <br />
      <br />
      As an addition, the simple <a target="_blank" href="https://rdml.org/rdes.html">Real-time PCR Data Essential 
      Spreadsheet Format (RDES)</a> was developped as an easy export format. In principle, it is a tab separated 
      text file with essential data and the cycles in columns and the wells in rows. RDES was first published in
      "<a href="https://pubmed.ncbi.nlm.nih.gov/36916500/" target="_blank">Disclosing quantitative RT-PCR raw 
      data during manuscript submission: a call for action in 2023 (Untergasser et al., 2023 Mol Oncol. 
      May;17(5):713-717. (doi: 10.1002/1878-0261.13418. or PMID: 36916500))</a>". 
    </p>

    <h5 id="License">1.3. License</h5>
    <p style="text-align: justify">
      Please read the <a class="plain" target="_blank" href="https://www.gear-genomics.com/terms">Terms of Use</a>
      before using the <a href="https://github.com/RDML-consortium/rdml-tools" target="_blank">RDML-Tools</a> and
      please obey to the requirements. The individual licenses are found in the respective
      <a href="https://github.com/RDML-consortium/rdml-tools" target="_blank">GitHub repositories</a>. The
      <a href="https://github.com/RDML-consortium/rdml-tools" target="_blank">RDML-Tools</a> mainly provide a web page
      access to the functions of the
      <a href="https://github.com/RDML-consortium/rdmlpython" target="_blank">RDML-Python library</a>, designed to
      handle RDML files and their dependencies. Users may choose to install the library as a
      <a href="https://github.com/RDML-consortium/rdmlpython-package" target="_blank">python package</a> calling
      "python -m pip install rdmlpython". All software is GPL licensed except the RDML-Python library being MIT
      licensed.
    </p>

    <h5 id="Citation">1.4. How to Cite</h5>
    <p style="text-align: justify">
      We request but do not require that use of this software be cited in publications as<br />
      <br />
      Web-based LinRegPCR: application for the visualization and analysis of (RT)-qPCR
      amplification and melting data.<br />
      Untergasser A, Ruijter JM, Benes V, van den Hoff MJB.<br />
      BMC Bioinformatics. 2021 Aug 24;22(1):398. doi: 10.1186/s12859-021-04306-1.<br />
      <br />
      The paper is available at
      <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04306-1" target="_blank">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04306-1</a>
      <br />
      <br />
      Source code available at
      <a href="https://github.com/RDML-consortium/rdml-tools" target="_blank">https://github.com/RDML-consortium/rdml-tools</a>.
    </p>

    <h5 id="LRP-Library">1.5. The Python Library rdmlpython and the Command Line</h5>
    <p class="regBottom" style="text-align: justify">The RDML-Tools are great to analyze single files and view the result. Behind the
      scenes the RDML-Tools do no calculations. They shape the input, hand it over to the server, the server does the
      calculations using the python library rdmlpython, collects the results and hands them to the RDML-Tools which
      display the results and allow to browse them interactively. If big files or a large number of files have to
      be analyzed or due to privacy reasons the data may not run on external servers, running rdmlpython locally
      via the command line interface might be an option. Programmers may use rdmlpython to handle RDML files in
      their own applications.
    </p>

    <p class="regBottom"><b>Installing rdmlpython</b></p>
    <pre>
    apt install python3 python3-pip

    pip3 install rdmlpython
    </pre>

    <b>The Command Line Interface</b><br />
    <p class="regBottom">Some functions of rdmlpython are available on the command line.</p>
    <p class="regBottom">Display the help with an overview of the possible commands:</p>
    <pre>
    rdmlpython -h
    rdmlpython --help
    </pre>

    <p class="regBottom">Display the version number of rdmlpython:</p>
    <pre>
    rdmlpython --version
    </pre>

    <p class="regBottom">Validate file data.rdml against its schema:</p>
    <pre>
    rdmlpython -v data.rdml
    rdmlpython --validate data.rdml
    </pre>

    <p class="regBottom">List all experiments in the file data.rdml:</p>
    <pre>
    rdmlpython -le data.rdml
    rdmlpython --listexperiment data.rdml
    </pre>

    <p class="regBottom">Select experiment "exp_1" and list all runs of this experiment in the file data.rdml:</p>
    <pre>
    rdmlpython -e exp_1 -lr data.rdml
    rdmlpython --experiment exp_1 --listrun data.rdml
    </pre>

    <p class="regBottom">Run RunAnalysis on file data.rdml and save the modified file in data_out.rdml and the
      analysis results to results.csv:</p>
    <pre>
    rdmlpython -lrp data.rdml -o data_out.rdml --saveResults results.csv
    rdmlpython --linRegPCR data.rdml --resultfile data_out.rdml --saveResults results.csv
    </pre>
    <p class="regBottom" style="text-align: justify">
      With the latter command, the analysis results are stored in a (large) tab-separated table results.csv that
      can be opened in spreadsheet programs like Excel or Calc. By default RunAnalysis will only work on the first run
      of the first experiment. You may select a different experiment with the -e option and a different run with the -r
      option. The option --excludeNoPlateau allows to exclude reactions with no plateau phase from the mean PCR efficiency
      calculation. The option --excludeEfficiency can be "outlier", "mean" or "include". With --excludeEfficiency
      "include" no reaction will be from mean PCR efficiency calculation based on its individual PCR efficiency. With
      --excludeEfficiency "mean" and the optional option --pcrEfficiencyExl 0.05 reactions with an individual PCR
      efficiency outside a 0.05 range from the mean PCR efficiency will be excluded from mean PCR efficiency
      calculation (this was the classic way). With --excludeEfficiency "outlier" only individual PCR efficiency
      outliers based on a Grubbs's test will be excluded from mean PCR efficiency calculation (recommended and the
      default option). The option --saveRaw raw_data.csv creates the table raw_data.csv with the raw fluorescence data.
      The option --saveBaslineCorr baseline_corrected.csv creates the table baseline_corrected.csv with the baseline
      corrected fluorescence data. See <a href="#RDML-RunAnalysis">3. RDML-RunAnalysis</a> for details on those options.
    </p>

    <b>Using the Python Library</b><br />
    <p class="regBottom"  style="text-align: justify">The rdmlpython library has many functions, please look at the
      <a href="https://github.com/RDML-consortium/rdmlpython/blob/master/rdml.py">source code of rdmlpython</a>
      for details.
    </p>
    <p class="regBottom">As an basic example create the file test.py with this content:</p>
    <pre>
    #! /usr/bin/env python

    import rdmlpython as rdml

    # Print the version
    vers = rdml.get_rdml_lib_version()
    print("Version " + vers)

    # Validate a file
    cli_validate = rdml.Rdml()
    cli_resValidate = cli_validate.validate(filename="test_2_raw_data.rdml")
    print(cli_resValidate)

    # Run RunAnalysis
    cli_linRegPCR = rdml.Rdml("test_2_raw_data.rdml")
    cli_expList = cli_linRegPCR.experiments()
    cli_exp = cli_expList[0]
    cli_runList = cli_exp.runs()
    cli_run = cli_runList[0]
    cli_result = cli_run.linRegPCR(updateRDML=True, saveResultsCSV=True, timeRun=True, verbose=True)
    cli_linRegPCR.save("test_2_data_out.rdml")
    with open("test_2_result.csv", "w") as cli_f:
        cli_f.write(cli_result["resultsCSV"])
    </pre>

    <p class="regBottom">Run the script with the <a href="static/bin/test_2_raw_data.rdml">downloaded file</a>:</p>
    <pre>
    python3 test.py
    </pre>


    <h5 id="Acknowledgements">1.6. Acknowledgement</h5>
    <p style="text-align: justify">
      We wish to thank Quinn Gunst and Jaco Hagoort (Department of Medical Biology, Amsterdam UMC, location, AMC,
      Amsterdam) for their advice and support while preparing the RDML-RunAnalysis program.
    </p>


    <h5 id="Tool-Introduction">2.1. Introduction</h5>
    <p style="text-align: justify">
      The RDML-Tools cover everything needet to analyze qPCR data. The general concept is to automate as 
      much as possible and archive reliable data independent of user input. To get the most out of RDML-Tools, 
      the user needs to complement a set of data, which will be described in the respective paragaphs below. For 
      example, an increase in fluorescence points to amplification. If the RDML-Tools were provided with the 
      respective sample type, they will be silent if the sample is a positive control or a unknown sample but 
      throw an error if it is a negative control. Naturally, the RDML-Tools will also thow an error if no 
      amplification occures in a positive control.
      <br />
      <br />
      The import of the data and the annotation of the data is cubersome, but usually a lab works with the same 
      machine and a usualy used set of targets. The targets need to be annotated only once and can be copied 
      over from well annotated files.
    </p>


    <h5 id="Import">2.2. Import Data into RDML</h5>
    <h5 id="Import-Introduction">2.2.1. Introduction into Importing Data</h5>
    <p style="text-align: justify">
      Although RDML and RDES are around for several decades, not all qPCR machines support one of the formats. 
      Therefore, the import remains a challenge today. In the following section all import options are explained.
    </p>

    <h5 id="Import-validRDMLfile">2.2.2. Import of a Valid RDML File</h5>
    <p style="text-align: justify">
      Ideally, a machine exports a valid RDML file, so no big modifications are needed. It is still recomendet 
      to perform some checks before the start. RDML-Tools require the RDML version 1.4, so the file may need to 
      be converted.
    </p>
    <ol>
    <li>Open <a href="https://www.rdml-tools.com/edit.html">RDML-Edit</a>, select the RDML file and press 
      "Upload File".</li>
    <li>Go back to the "Main" tab. Check the version in "RDML File Information".</li>
    <li>If the version was not 1.4, click on "Enable Edit Mode" on the top, then the "Migrate to RDML Version ..." 
      buttons appear. Click them one by one inceasing the version of the file to 1.4.</li>
    <li>In the "Links to other RDML tools" section, click on the link to RDML-Validate. It should state that the
      file is valid. Else continue with <a href="#Import-invalidRDMLfile">Import of a invalid RDML file</a>.
    </li>
    <li>Now the file is ready. Do not forget to save it by pressing "Save File".</li>
    </ol>

    <h5 id="Import-invalidRDMLfile">2.2.3. Import of a Invalid RDML File</h5>
    <p style="text-align: justify">
      Some qPCR machines do export RDML files, but the files are not valid. For some machines RDML-Tools can 
      fix this errors archiving a valid RDML file.
    </p>
    <ol>
    <li>Open <a href="https://www.rdml-tools.com/edit.html">RDML-Edit</a>, select the RDML file and press 
      "Upload File". A error mentioning the invalid RDML file might appear.</li>
    <li>Go back to the "Main" tab and click on "Enable Edit Mode" on the top. Then green buttons appear. Click 
      on "Repair RDML File" and "Recreate Lost IDs".</li>
    <li>Check the version in "RDML File Information". If the version was not 1.4, then click the "Migrate to RDML 
      Version ..." buttons one by one inceasing the version of the file to 1.4.</li>
    <li>In the "Links to other RDML tools" section, click on the link to RDML-Validate. If it does not state 
      "RDML file is valid.", send us the RDML file for further inspection.</li>
    <li>Now the file is ready. Do not forget to save it by pressing "Save File".</li>
    </ol>

    <h5 id="Import-validRDESfile">2.2.4. Import of a Valid RDES File</h5>
    <p style="text-align: justify">
      Some machines do export RDES files. In this case RDML-Edit can be used to create an RDML file. Be aware, that 
      amplification data and melting curve data are saved in seperate RDES files.
    </p>
    <ol>
    <li>Open <a href="https://www.rdml-tools.com/edit.html">RDML-Edit</a> and press the
      "Create New File" button.</li>
    <li>RDML-Edit switches to the "Experiments" tab. Click on "Create New Experiment".</li>
    <li>Give the experiment a name and press "Save Changes".</li>
    <li>Now the experiment excists and a new run can be created, press "New Run".</li>
    <li>Give the run a name.</li>
    <li>Select the plate format. The pedefined buttons give shortcuts for the most common formats. Do also check
      the "Reaction Volume" fits your experiment in &micro;l.</li>
    <li>Select the respective RDES files for upload at the "Import Amplification RDES" and "Import Melting RDES" 
      fields.</li>
    <li>Press "Save Changes" and evaluate the import message. Check that the "Number of Reactions" field of the 
      experiment fits the expectations and no reactions are lost. If it fails, read the next paragraphs for some 
      tips on how to continue.</li>
    <li>Repeat the steps above if more runs need to be imported.</li>
    <li>Now the file is ready. Do not forget to save it by pressing "Save File".</li>
    </ol>

    <h5 id="Import-RDES-Introduction">2.2.5. Introduction Into RDES</h5>
    <p style="text-align: justify">
      If RDES files need to be repaired or CSV files need to be converted, a understanding of the RDES file format is 
      required. It consists of seven columns left of the fluorescence block of which the first six need to be filled.
      The sample type may be one of "unkn", "std", "pos", "nac", "ntc", "nrt" or "ntp". The target type may be "toi" 
      or "ref". Sample and target may be any text, but be aware that technical replicates must have identical sample 
      and target information. Find below an example with amplification data. Melting data will have temperatures 
      instead of cycle in the heading column and colum 7 must be "Tm", but can be empty like the Cq column. Correctly 
      formated RDES files require tab as column separator, linux newlines and dot as decimal separator. See 
      <a target="_blank" href="https://rdml.org/rdes.html">Real-time PCR Data Essential Spreadsheet Format (RDES)</a>
      for more information about the format.
    </p>

    <table class="table table-bordered table-hover table-condensed">
      <thead>
        <tr>
          <th>Well</th><th>Sample</th><th>Sample Type</th><th>Target</th><th>Target Type</th><th>Dye</th><th>Cq</th><th>1</th><th>2</th><th>3</th><th>...</th><th>44</th><th>45</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>A1</td><td>sam 1</td><td>unkn</td><td>ANF</td><td>toi</td><td>unkn_dye</td><td> </td><td>167.61</td><td>171.84</td><td>175.77</td><td>...</td><td>635.11</td><td>634.88</td>
        </tr>
        <tr>
          <td>A2</td><td>sam 1</td><td>unkn</td><td>EGFP</td><td>toi</td><td>unkn_dye</td><td> </td><td>185.18</td><td>194.59</td><td>201.04</td><td>...</td><td>925.12</td><td>923.84</td>
        </tr>
        <tr>
          <td>A3</td><td>sam 1</td><td>unkn</td><td>GAPDH</td><td>toi</td><td>unkn_dye</td><td> </td><td>179.11</td><td>182.69</td><td>185.65</td><td>...</td><td>735.28</td><td>739.5</td>
        </tr>
        <tr>
          <td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td>
        </tr>
        <tr>
          <td>H11</td><td>neg</td><td>unkn</td><td>RT-</td><td>toi</td><td>unkn_dye</td><td> </td><td>185.15</td><td>188.06</td><td>190.16</td><td>...</td><td>712.36</td><td>718.66</td>
        </tr>
        <tr>
          <td>H12</td><td>neg</td><td>unkn</td><td>RT-</td><td>toi</td><td>unkn_dye</td><td> </td><td>170.7</td><td>180.56</td><td>183.9</td><td>...</td><td>653.92</td><td>660.56</td>
        </tr>
      </tbody>
    </table>

    <h5 id="Import-Raw-Data-Warning">2.2.6. A Note on Raw Data</h5>
    <p style="text-align: justify">
      The RDML-Tools may throw a warning if negative fluorescence values are imported or used. Actually, fluorescence 
      values can never be negative, as zero means no light is emitted. Usually negative values occur if a baseline 
      correction was applied to the raw fluorescence data.
      <br /><br />
      Most machines perform a baseline correction based on the first 3(5)-10(15) cycles and fit a linear function, 
      which is subtracted from the raw fluorescence data. If only a constant value was subtracted, the calculations 
      are not affected and the constant value is removed. If the subtracted linear function has also an inclination, 
      RDML-Tools will not be able to calculate the correct PCR efficiencies. There is no way to undo the curve 
      based baseline correction.
      <br /><br />
      If this error occurs, check if the software of the qPCR machine allows to switch off the baseline correction. 
      Contact the vendor and ask for help - and maybe also ask why there is no RDML or RDES export, as this will make 
      life much easier. Up to then, there is nothing you can do about it. You may continue using RDML-Tools, but be 
      warned that the PCR efficiencies might be affected. You may try DNA dilution standards and see if 
      RDML-ExperimentAnalysis reports similar PCR efficiencies calculated by the curves and calculated by dilution.
    </p>

    <h5 id="Import-repairRDESfile">2.2.7. Repair a RDES File</h5>
    <p style="text-align: justify">
      RDML-Edit demands correctly formated RDES files, with tab as column separator, linux newlines and dot as 
      decimal separator. Opening the file in Speadsheed programms can easily mess this up and making the import 
      impossible. This can be easily fixed in RDES-TableShaper.
    </p>
    <ol>
    <li>Open <a href="http://localhost:1234/tableshaper.html">RDES-TableShaper</a> select the RDES file and press 
      "Upload File".</li>
    <li>Check that the file contained the intended data below the "Table after import". Confirm that the colums 
      and rows were recognized and labled with numbers in the grey fields.</li>
    <li>Depending on imported data, click on "RDES Amplification" or "RDES Meltcurve".</li>
    <li>Go to the "Save" tab an inspect the RDES file below. All rows which should be imported must have a 
      green background.</li>
    <li>In case information is missing, the information can be added using the "Complement columns 1 to 6" 
      section. Sample and target may be any text, but be aware that technical replicates must have identical sample 
      and target information. Optionally all samples can be labled "sample1" and all targets "target1". Then the 
      correct annotation must be provided in RDML-Edit later.
    <li>Now the file is ready. Save it by pressing "Save table file in RDES format" on the top.</li>
    <li>Repeat the procedure to have amplification and melting RDES files.</li>
    <li>Continue creating an RDML file following <a href="#Import-validRDESfile">Import of a Valid RDES 
      file</a></li>
    </ol>

    <h5 id="Import-knownCSVfile">2.2.8. Import of a Known CSV File</h5>
    <p style="text-align: justify">
      Some machines do not export RDML or RDES files. In this case the export is more challenging as it has to be taken 
      care that the raw fluorescent measurments of each cycle are exported. RDML-Tools do only read csv formated 
      text files. Most qPCR machines export Excel files. Please convert them by opening in Excel (Microsoft) or Calc 
      (Libre Office), selecting the table with the amplification or melting data and save as tab separated text file, 
      ideally in UTF-8 txt.
    </p>
    <ol>
    <li>Open <a href="http://localhost:1234/tableshaper.html">RDES-TableShaper</a> select the RDES file and press 
      "Upload File".</li>
    <li>Check that the file contained the intended data below the "Table after import". Confirm that the colums 
      and rows were recognized and labled with numbers in the grey fields.</li>
    <li>Go to the "Modify" tab an select the qPCR format of your machine from the "Load predefined setting" 
      pulldown menue.</li>
    <li>Go to the "Save" tab an inspect the RDES file below. All rows which should be imported must have a 
      green background.</li>
    <li>In case information is missing, there are several options. 
      |<ul style="list-style-type:disc;">
        <li>The information can be added in this tab using the "Complement columns 1 to 6" section. Single wells  
          like A7 or squares B2-D7 can be used. Sample and target may be any text, but be aware that technical 
          replicates must have identical sample and target information.</li>
        <li>The current file can be downloaded by pressing "Save table file in RDES format" on the top. Open 
          the file in Excel (Microsoft) or Calc (Libre Office) and complement the missing data and save the file. 
          It is recommended to import the file once more as described in <a href="#Import-repairRDESfile">Repair 
          a RDES file</a> to get the column seperators and line endings right.</li>
        <li>Lable all samples "sample1" and all targets "target1". Then the correct annotation must be provided 
          using RDML-Edit and other tools later. This could be more convenient.</li>
      </ul>
    <li>Now the file is ready. Save it by pressing "Save table file in RDES format" on the top.</li>
    <li>Repeat the procedure to have amplification and melting RDES files.</li>
    <li>Continue creating an RDML file following <a href="#Import-validRDESfile">Import of a Valid RDES 
      file</a></li>
    </ol>

    <h5 id="Import-unknownCSVfile">2.2.9. Import of a Unknown CSV File</h5>
    <p style="text-align: justify">
      If your qPCR machine is not listed in "Load predefined settings" on the "Modify" tab of RDES-TableShaper, 
      please contact us and send us the file, so we can add it to the list. Of course, you may also create an 
      import template yourself. Therefore, the layout of the table has first to be evaluated. Open the file in 
      a spreadsheet program or an editor and evaluate the columns and rows. It may have the same layout like 
      RDES, the wells/reachtions are given in rows, the cycles in the columns. This would require later "Keep 
      table shape" later in the protocol:
    </p>
    <table class="table table-bordered table-hover table-condensed">
      <thead>
        <tr>
          <th> </th><th>1</th><th>2</th><th>3</th><th>...</th><th>44</th><th>45</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>A1</td><td>167.61</td><td>171.84</td><td>175.77</td><td>...</td><td>635.11</td><td>634.88</td>
        </tr>
        <tr>
          <td>A2</td><td>185.18</td><td>194.59</td><td>201.04</td><td>...</td><td>925.12</td><td>923.84</td>
        </tr>
        <tr>
          <td>A3</td><td>179.11</td><td>182.69</td><td>185.65</td><td>...</td><td>735.28</td><td>739.5</td>
        </tr>
        <tr>
          <td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td>
        </tr>
        <tr>
          <td>H11</td><td>185.15</td><td>188.06</td><td>190.16</td><td>...</td><td>712.36</td><td>718.66</td>
        </tr>
        <tr>
          <td>H12</td><td>170.7</td><td>180.56</td><td>183.9</td><td>...</td><td>653.92</td><td>660.56</td>
        </tr>
      </tbody>
    </table>
    <p style="text-align: justify">
      <br /><br />
      Alternatively, it may be transposed with cycles as rows and the wells/reachtions as columns. This would 
      require later "Flip columns and rows" later in the protocol:
    </p>
    <table class="table table-bordered table-hover table-condensed">
      <thead>
        <tr>
          <th></th><th>A1</th><th>A2</th><th>A3</th><th>...</th><th>H11</th><th>H12</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>1</td><td>167.61</td><td>185.18</td><td>179.11</td><td>...</td><td>185.15</td><td>170.7</td>
        </tr>
        <tr>
          <td>2</td><td>171.84</td><td>194.59</td><td>182.69</td><td>...</td><td>188.06</td><td>180.56</td>
        </tr>
        <tr>
          <td>3</td><td>175.77</td><td>201.04</td><td>185.65</td><td>...</td><td>190.16</td><td>183.9</td>
        </tr>
        <tr>
          <td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td>
        </tr>
        <tr>
          <td>44</td><td>635.11</td><td>925.12</td><td>735.28</td><td>...</td><td>712.36</td><td>653.92</td>
        </tr>
        <tr>
          <td>45</td><td>634.88</td><td>923.84</td><td>739.5</td><td>...</td><td>718.66</td><td>660.56</td>
        </tr>
      </tbody>
    </table>
    <p style="text-align: justify">
      <br /><br />
      Or it may come as a list, were each measurement is a row. This would require later 
      "Create table from list" later in the protocol:
    </p>
    <table class="table table-bordered table-hover table-condensed">
      <thead>
        <tr>
          <th>Well</th><th>Cycle</th><th>Fluor</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>A1</td><td>1</td><td>167.61</td>
        </tr>
        <tr>
          <td>A1</td><td>2</td><td>171.84</td>
        </tr>
        <tr>
          <td>A1</td><td>3</td><td>175.77</td>
        </tr>
        <tr>
          <td>...</td><td>...</td><td>...</td>
        </tr>
        <tr>
          <td>H12</td><td>44</td><td>653.92</td>
        </tr>
        <tr>
          <td>H12</td><td>45</td><td>660.56</td>
        </tr>
      </tbody>
    </table>
    <p style="text-align: justify">
      <br /><br />
      With this information, a custom converter template can be created: 
    </p>
    <ol>
    <li>Take care the fluorescence data have been saved as text file and an editor like notepad is able to read 
      the file.</li>
    <li>Open <a href="http://localhost:1234/tableshaper.html">RDES-TableShaper</a> select the text file and press 
      "Upload File".</li>
    <li>Now the table separators have to be identified on the "Load" tab. It should work out automatically, if 
      not, the separator can be selected manually in the "Select column separator for import" section. Check 
      that the table on the bottom of the "Load" tab breaks up in columns and each column has a number in the 
      grey row on top.</li>
    <li>Go to the "Modify" tab. In the section "Settings to reshape the table" select "Amplification" or 
      "Meltcurve" based on the type of your input data.</li>
    <li>Give a name to the converter in "Settings identifier"</li>
    <li>Select "Keep table shape", "Flip columns and rows" or "Create table from list" based on the table layout 
      determined as mentioned on top of this protocol.</li>
    <li>Select "Replace all commas by dots" if the data use a comma as decimal separator.</li>
    <li>Now the table should appear reshaped in the bottom of this tab. It has numbers for rows and columns with 
      a grey background. This numbers are required to fill the following fields.</li>
    <li>In the "Settings to extract the fluorescence data block" section the fluorescence data have to be selected 
      by providing the numbers of columns and rows on top or left of it. In the list format, the column with the 
      fluorescence data has to be provided.</li>
    <li>In the "Settings to extract data" section the row with the cycle or temp. has to be selected. RDES can 
      collect additional information. This data can also be collected from the file if the information is present.
      Each field can be modified by a regular expression, if desired. Regular expressions are very powerful but 
      are out of the scope of this manual. If the default (.*) needs to be modified, ask any informatician for 
      help (it will be an easy task) or dive into Javascript capturing regular expressions tutorials.
    </li>
    <li>Go to the "Save" tab and inspect the table on the bottom. It should be filled with all the information 
      selected before. Check that the fluorescence data are there, cycles or temp. were correctly read in the 
      first row.
    </li>
    <li>If this worked well, go back to the "Modify" tab and save your custom import using the "Save settings 
      JSON file" button on the top. If other data have to be imported, the steps before can ve skipped and this 
      file can be uploaded using "Load settings JSON file" on the left instead. Then go back to the "Save" tab.</li>
    <li>Usually, some of the first 6 columns are not filled after that step. Without this information, the data 
      can not be imported as described at <a href="#Import-validRDESfile">Import of a Valid RDES file</a> and 
      the background will not be green. There are several options at this step. </li>
    <li>In case information is missing, there are several options. 
      |<ul style="list-style-type:disc;">
        <li>The information can be added in this tab using the "Complement columns 1 to 6" section. Single wells  
          like A7 or squares B2-D7 can be used. Sample and target may be any text, but be aware that technical 
          replicates must have identical sample and target information.</li>
        <li>The current file can be downloaded by pressing "Save table file in RDES format" on the top. Open 
          the file in Excel (Microsoft) or Calc (Libre Office) and complement the missing data and save the file. 
          It is recommended to import the file once more as described in <a href="#Import-repairRDESfile">Repair 
          a RDES file</a> to get the column seperators and line endings right.</li>
        <li>Lable all samples "sample1" and all targets "target1". Then the correct annotation must be provided 
          using RDML-Edit and other tools later. This could be more convenient.</li>
      </ul>
    <li>Now the file is ready. Save it by pressing "Save table file in RDES format" on the top.</li>
    <li>Repeat the procedure to have amplification and melting RDES files.</li>
    <li>Continue creating an RDML file following <a href="#Import-validRDESfile">Import of a Valid RDES 
      file</a></li>
    </ol>


    <h5 id="RDML-Edit">2.3. Supplement Remaining Data using RDML-Edit</h5>
    <p style="text-align: justify">
      If the import as described in <a href="#Import">Import Data into RDML</a> worked out, a vaild RDML file version 
      1.4 should have been created. Although it is valid, some information should be added to get the most of the 
      RDML-Tools. Besides RDML-Edit dedicated tools like RDML-RunEdit, RDML-AnnotationEdit and RDML-Merge help with 
      special functionality. 
    </p>
    <p style="text-align: justify">
      RDML-Edit is a powerful tool to create, view and edit RDML files. RDML files supports many optional parameters and
      each parameter has many sub-parameters, of which many may not yet be used in the current file. In RDML a target 
      for example is described only once in the targets list with all its futher information and thereafter is only 
      identified by its unique id. In the data elements of each reaction the target is then referenced by its id.
    </p>
    <p style="text-align: justify">
      Be aware that RDML-Edit by default only allows viewing of files. RDML support many optional parameters to
      describe a qPCR or digital PCR reaction. RDML-Edit will hide all parameters which were not entered by the user.
      <b>To modify RDML files you have to enable the edit mode using the button on the "main" tab.</b> Enabling the edit
      mode will switch RDML-Edit into a different mode and enable many buttons and options to modify each element or
      create new ones. RDML does support different versions with different functionality. RDML-Edit can migrate
      between different versions on the "Main" tab if the edit mode is activated. Digital PCR is only supported from
      version 1.3 on. The modified RDML-files must be saved and downloaded on the "Main" tab as they will be lost once 
      the browser is closed.
    </p>
    <p style="text-align: justify">
      Lets assume all samples were labled "sample1" and all targets "target1" and only the fluorescence data of the 
      amplification and the melting curve were imported. Lets repair the run setup and add the rrequired additional 
      information step by step. First the RDML file is loaded and dye information is added:
    </p>
    <ol>
    <li>Open <a href="https://www.rdml-tools.com/edit.html">RDML-Edit</a>, select the RDML file and press 
      "Upload File".</li>
    <li>Go to the "Main" tab and check the version in "RDML File Information". If the version was not 1.4, then 
      click the "Migrate to RDML Version ..." buttons one by one inceasing the version of the file to 1.4.</li>
    <li>Click on "Enable Edit Mode" on the top. Then green buttons appear.</li>
    <li>Go to the "Dye" tab and press the "Create New Dye" button.</li>
    <li>Fields with an orange background must be provided with information. Give the dye a ID like "SYBR I" 
      and press "Save Changes".</li>
    <li>The dye "SYBR I" excists now. Press "Edit" to add more information.</li>
    <li>Select the proper "Fluorescence Reporter". This is essential for the calculation of Ncopy. If not provided, 
      "non-saturating DNA binding Dye" will be used by default.</li>
    <li>If a concentration of the dye is known (not of a hybridisation probe, that goes somewhere different), it can 
      be added as well.</li>
    <li>Press "Save Changes" and repeat this for all other dyes.</li>
    </ol>
    <p style="text-align: justify">
      RDML is a flexible format. On the "Documentation" tab information text can be stored and attached by its ID at 
      several other positions. For example the freezing of the samples could be described and attached at all the 
      samples which were frozen that way. Similarily, the "Experimenters" or the "Cycling Conditions" can be 
      mentioned and attached to the respective runs. Also this is recommended for documentation purposes, it has no 
      impact on the calculations of the RDML-Tools.
    </p>
    <p style="text-align: justify">
      The target information needs to be added next. The target information are essential for documentation and for 
      reporducibility and are required for the correct calculations of the RDML-Tools:
    </p>
    <ol>
    <li>Assuming the RDML file is still opened from the previous step, to the "Targets" tab and press the "Create 
      New Target" button.</li>
    <li>Provide the target with a understandable and unique "ID".</li>
    <li>Select a "Type". Use "toi - target of interest" for all targets that should be quantified. Use "ref - 
      reference target" if this target should be used as a reference gene to correct the gene expression of other 
      genes. If it is not labled here as "ref", it can not be used in the "Quantify" tab or "geNorm" tab of 
      RDML-ExperimentAnalysis later.</li>
    <li>Select the suitable dye used for this target. If the same target is used with different dyes, a second 
      target with a different ID needs to be created.</li>
    <li>Provied the primer sequences and the concentrations of the forward and reverse primer, the hydrolysis probe 
      as "probe1" if present and the amplicon sequence. The amplicon length is essential for the correct calculation 
      of non-saturationg dyes, the primer concentration for probes. Primer concentrations are usually in the range of 
      100-700nMol/l, if not provided, 250nMol/l is used. If the amplicon sequence can not be provided or is not 
      available, provide the length of the amplicon using "Amplicon Seq. Tools". It will create a string of Ns of 
      appropriate length.</li>
    <li>Provide the "Amplicon MeltingTemperature" if DNA binding dyes are used. This should be determined in a pilot 
      experiment for each target. Once provided here, RDML-RunAnalysis can evaluate melting curves for the presence of 
      the expected amplicon and warn if artifacts are detected.</li>
    <li>Press "Save Changes" and repeat this for all other targets.</li>
    </ol>
    <p style="text-align: justify">
      The sample information needs to be added next. The sample information are essential for the correct calculations 
      of the RDML-Tools:
    </p>
    <ol>
    <li>Assuming the RDML file is still opened from the previous step, to the "Samples" tab and press the "Create 
      New Sample" button.</li>
    <li>Provide the target with a understandable and unique "ID".</li>
    <li>Select a "Type". This is essential for RDML-Tools an it will be used to perform validy checks. Positive controls 
      and standards must archive an amplification. Negative controls may not and unknown samples may both. The sample 
      type will result in different, apropriate handling of the sample.</li>
    <li>Select "Double Stranded" yes for plasmids, genomic DNA and PCR products, no for cDNA or other single stranded 
      DNA. This parameter is essential for the correct calculation of Ncopy. Speaking of "one copy", a single copy of 
      double stranded DNA (gnomic DNA) or a single standed DNA (cDNA) could be meant.</li>
    <li>Press "Save Changes".</li>
    <li>If the sample is part of a standard, the sample type "std" should be selected and the concentration should be 
      given by pressing the "Add Quantity" button. Be aware that here the concentration should be given in copies per 
      microliter of the reaction mix in the tube. This is essential if a standard should be used to quantify or a 
      dilution standard should be evaluated in the "Quantify" tab of RDML-ExperimentAnalysis</li>
    <li>Repeat this for all other samples.</li>
    <li>Save the edited RDML file. Go to the "Main" tab and press the "Save File" button.</li>
    </ol>
    <p style="text-align: justify">
      Now the dyes, samples and targets are provided, but not used in the runs. Now the experiment ID and the run 
      ID can be adapted in the "Experiments" tab. Check that the correct PCR format was selected and attach the cycling 
      conditions to the respective run. This is advisable for documentation and presentation of the data, but has no 
      impact on the calculations. To attach the information to the run, <a href="#RDML-RunEdit">RDML-RunEdit</a> 
      is required.
    </p>


    <h5 id="RDML-RunEdit">2.4. Annotating a Run in RDML-RunEdit</h5>
    <p style="text-align: justify">
     RDML-RunEdit is a tool to view single runs of a RDML files. Once the RDML file was loaded, the desired experiment
     and run have to be selected. RDML-RunView will display the plate setup with the errors and the notes editable. 
     The user can adapt the target, sample and volume information and adapt or delete incorrect errors and notes or 
     introduce errors orrured during qPCR preperation. Emply wells may be deleted.
    </p>
    <p style="text-align: justify">
      If a complete plate was imported, it is advisble to delete the empty wells for a better overview:
    </p>
    <ol>
    <li>Open <a href="https://www.rdml-tools.com/runedit.html">RDML-RunEdit</a>, select the RDML file and press 
      "Upload File".</li>
    <li>Go to the "RunEdit" tab and select the correct experiment and run on the top.</li>
    <li>It is recommended to open the same run in 
      <a href="https://www.rdml-tools.com/runanalysis.html">RDML-RunAnalsyis</a> in a seperate window and inspect the 
      plate on the "RunView" tab. Using the "Select Sample by" and providing "Range" like A1-B6;C7-D8, the 
      fluorescence of the wells can be evaluated and removed in RDML-RunEdit in a different window.</li>
    <li>Select the "Reaction" for the change. It accepts single wells like A1 or ranges like A1-D3. For convenience, this 
      field is filled by clicking in one or one cell and a second cell in the grid of the run above.</li>
    <li>Click "Delete Reaction Data" to remove the well.</li>
    <li>Repeat this for all other empty wells.</li>
    </ol>

    <p style="text-align: justify">
      Then the targets and samples can be attached:
    </p>
    <ol>
    <li>Assuming the RDML file is still opened from the previous step, go to the "RunEdit" tab.</li>
    <li>Select the "Reaction" for the change. It accepts single wells like A1 or ranges like A1-D3. For convenience, this 
      field is filled by clicking in one or one cell and a second cell in the grid of the run above.</li>
    <li>Select the sample an click "Replace Sample".</li>
    <li>Check that the range still fits.</li>
    <li>Select the target an click "Replace Target".</li>
    <li>Repeat this for all other reactions.</li>
    </ol>

    <p style="text-align: justify">
      If the reaction volume was not set at import or is different within a plate, it can be adapted as well:
    </p>
    <ol>
    <li>Assuming the RDML file is still opened from the previous step, go to the "RunEdit" tab.</li>
    <li>Select the "Reaction" for the change. It accepts single wells like A1 or ranges like A1-D3. For convenience, this 
      field is filled by clicking in one or one cell and a second cell in the grid of the run above.</li>
    <li>Set the "Reaction Volume in Microliter".</li>
    <li>Click "Save Reaction Volume".</li>
    <li>Repeat this for all other reactions.</li>
    <li>Save the edited RDML file. Press the "Save RDML" button on top.</li>
    </ol>

    <p style="text-align: justify">
      RDML-RunEdit is also an efficient way to edit notes and excluded text in the run. Be aware that a reaction with any 
      excluded text will be removed from all further calculations. It is good practice to give the reason for exclusion 
      in the excluded text or write any special obeservations into the note text.
    </p>
    <ol>
    <li>Open <a href="https://www.rdml-tools.com/runedit.html">RDML-RunEdit</a>, select the RDML file and press 
      "Upload File".</li>
    <li>Go to the "RunEdit" tab and select the correct experiment and run on the top.</li>
    <li>Select the "Reaction" for the change. It accepts single wells like A1 or ranges like A1-D3. For convenience, this 
      field is filled by clicking in one or one cell and a second cell in the grid of the run above.</li>
    <li>Modify or delete the "Excluded" and/or the "Notes" text.</li>
    <li>Update the file by pressing the respective buttons. "Replace" will overwrite the existing text or delete it if the 
      text is empty. "Append" attaches the text to the text which is already present.</li>
    <li>Repeat this for all other reactions.</li>
    <li>Save the edited RDML file. Press the "Save RDML" button on top.</li>
    </ol>

    <p style="text-align: justify">
      RDML-RunEdit can also delete all calculation results, notes and errors present in a run to allow a fresh analysis.
    </p>


    <h5 id="RDML-Merge">2.5. Copy Annotation Using RDML-Merge</h5>
    <p style="text-align: justify">
      RDML works with elements that refer other elements. For example, the reactions in an experiment refer to samples and 
      targets. The targets refer to an dye. The RDML-Tools require quite extensive annotation as described in 
      <a href="#RDML-Edit">Supplement Remaining Data using RDML-Edit</a>. Usually, the same targets, dyes and cycling 
      protocols are used over and over again. RDML-Merge helps by complementing one RDML file (base) with parts of 
      a second RDML file (add). Once both RDML files were loaded, the desired elements have to be selected. RDML-Merge 
      will display an overview of both files with the contents of the base file on the left and the add file on the 
      right. The user can click on the respective buttons to copy individual or all elements over. This also allows 
      to combine several runs from different files within one experiment.
    </p>
    <p style="text-align: justify">
      If <a href="https://www.rdml-tools.com/merge.html">RDML-Merge</a> merge copies over an experiment, the dependencies 
      will be copied as well. If such a depending element with the same ID is present in the base RDML file and the 
      RDML file to add, the button with (-) will keep the dependencies of the base RDML file while the button with 
      (+) will overwrite the existing elements with the dependencies from the RDML file to add. 
    </p>
    <p style="text-align: justify">
      The button "Save Base RDML File" on the top left of the "Merge" tab will save the modified file. 
    </p>


    <h5 id="RDML-AnnotationEdit">2.6. Annotate Experiments Using RDML-AnnotationEdit</h5>
    <p style="text-align: justify">
      RDML can only handle technical replicates, reactions with the same sample ID and the same target ID are 
      considered technical replicates. All other samples must have unique sample IDs. Experiments require the 
      grouping of samples into different experimental groups or it may be of use to attach further information 
      to individual samples like "sex" or "age". In RDML an annotation element is present, which works with 
      keys and values. RDML-AnnotationEdit modifies this annotation.
    </p>
    <ol>
    <li>Open <a href="https://www.rdml-tools.com/annotationedit.html">RDML-AnnotationEdit</a>, select the RDML file
       and press "Upload File".</li>
    <li>Go to the "AnnotationEdit" tab.
      <ul style="list-style-type:disc;">
        <li>A annotation csv file can be downloaded by pressing "Export Annotation Data" on the top left. This file
          can be opend in Excel (Microsoft) or Calc (Libre Office). Additional annotation can be provided by 
          adding a key in the first row of an additional column. Then add the values for each sample in this 
          new column. Save the file and upload it by selectiong it with "Import Annotation Data" and click 
          the "Import Annotation Data" button.
        </li>
        <li>With the "Create New Annotation from two existent Annotations" section a key and a value can be 
          given and the corresponding samples selected. By pressing the button "Create Annotations", the annotations
          are created.
        </li>
      </ul>
    </li>
    <li>Repeat this for all other annotations.</li>
    <li>Save the edited RDML file. Press the "Save RDML" button on top right.</li>
    </ol>

    <p style="text-align: justify">
      RDML-ExperimentAnalysis can only handle one key a time. If the experiment requires combinations of keys, 
      for example "sex" and "age", RDML-AnnotationEdit can create an additional key with this combination.
    </p>
    <ol>
    <li>Open <a href="https://www.rdml-tools.com/annotationedit.html">RDML-AnnotationEdit</a>, select the RDML file
       and press "Upload File".</li>
    <li>Go to the "Create New Annotation from two existent Annotations" section on the "AnnotationEdit" tab.</li>
    <li>Give new key for the combined element in "Combined Property Name".</li>
    <li>Provide an excisting key for the left part of the combined element in "Property Left".</li>
    <li>The text provided in "Connect with" will be placed between the left and the right key.</li>
    <li>Provide an excisting key for the right part of the combined element in "Property Right".</li>
    <li>Click "Create Combined Annotation" to merge the keys and create a new one.</li>
    <li>Save the edited RDML file. Press the "Save RDML" button on top right.</li>
    </ol>
    <p style="text-align: justify">
      RDML-AnnotationEdit can also be used to rename properties (keys) and values using the top sections. 
    </p>


    <h5 id="RDML-RunAnalysis">2.7. Analyze a Run with RDML-RunAnalysis</h5>
    <p style="text-align: justify">
      RDML-RunAnalysis displays and analyzes melting curves, finds peaks and calculates the amplicon melting 
      temperature. It also displays and analyzes amplification curves, calls third derivative zero (TD0) 
      positions which RDML-Tools use instead of Cq values, calculates PCR efficiencies and Ncopy values.
    </p>
 
    <h5 id="RDML-RunAnalysis-Melting">2.7.1. Melting Curve Analysis with RDML-RunAnalysis</h5>
    <p style="text-align: justify">
      Primers might bind additionally to unintended positions in the DNA and amplify additional amplicons, 
      named artifacts. It is not possible to deduce the presence of an artifact product based on the shape of 
      the amplification curve. DNA-binding dyes, like SYBR-Green I and LCGreen bind unspecifically to DNA and 
      do not discriminate between the correct product and artifacts. The artifacts can only be detected by 
      recording a melting curve, where after completion of the amplification cycles the machine gradually heats 
      up the reactions from 60°C to 95°C and records the fluorescence. If the melting temperature (Tm) of the 
      product is reached, the fluorescence drops and the negative first derivative of the melting curve will 
      show a melting peak for each product. The melting temperature depends on amplicon length and sequence, 
      the reaction buffer composition. Therefore, correct products and artifacts can be discriminated based 
      on their melting temperature.
    </p>
    <p style="text-align: justify">
      RDML-RunAnalysis detect the correct products and artifacts automatically on the "MeltingCuveAnalysis" tab, 
      provided that the expected melting temperatures of the targets are given as described in 
      <a href="#RDML-RunEdit">Annotating a Run in RDML-RunEdit</a>. A reaction with several products is reported 
      and excluded from further analysis. 
    </p>
    <ol>
    <li>Open <a href="https://www.rdml-tools.com/runanalysis.html">RDML-RunAnalysis</a>, select the RDML file and press 
      "Upload File".</li>
    <li>Go to the "RunView" tab and select the desired experiment and run.</li>
    <li>Go to the "MeltingCuveAnalysis" tab and click "Run Meltcurve".</li>
    <li>A table is calculated with a row for each reaction, grouped by target. A set of columns is created for 
      each peak. The first set of columns allways displays the information about the expected peak. Evaluate 
      the expected peaks, the excluded and note messages.</li>
    <li>This table can be copied to the clipboard by clicking "Copy table to clipboard" and pasted in a spreadsheet 
      program or downloaded TSV file by clicking "Save table as CSV" for later use. </li>
    <li>A Double click on the row opens an individual curve in the "RunView" tab and allows the inspection of 
      the melting peaks.</li>
    <li>In the overview on the "RunView" tab, only one target can be selected by setting "Select Sample by" to 
      "Target" and choosing the desired target with "and". The thin black line shows the expected melting 
      temperature, the grey area highlights the acceptable range for the melting peak.</li>
    <li>The curves should be inspected and reactions with artifacts should be excluded from further analysis.</li>
    <li>All reactions which are marked as excluded by any text in this column are ignored in further analysis. 
      On the RunView tab the excluded and notes field can be modified or deleted, provided a single reaction is 
      selected. Be aware that the table on the "Melting Curve Analysis" tab is not updated. Alternativley, this 
      can be done by clicking the link below "Edit a single Run in RunEdit" on the "Main" tab and following the 
      instructions of <a href="#RDML-RunEdit">RDML-RunEdit</a>.</li>
    <li>Save the analyzed RDML file by clicking "Save RDML" on the "MeltingCuveAnalysis" tab.</li>
    </ol>

    <h5 id="RDML-RunAnalysis-Amplification">2.7.2. Amplification Curve Analysis with RDML-RunAnalysis</h5>
    <p style="text-align: justify">
      The exponential amplification of an PCR amplicon starts from the beginning of the qPCR run. Due to the 
      minute amounts at start, the fluorescence of the product is below the background noise of the machine and 
      can only be detected when sufficient amounts were amplified to pass background fluorescence. Shortly after 
      the components in the reaction mix, usually the non-saturating dye or the primers get limiting and the 
      exponential phase of the amplification ends. The "AmplificationCurveAnalysis" tab can calculate a TD0, the 
      point when the exponential phase ended, a PCR efficieny per target and the Ncopy value for each reaction.
    </p>
    <p style="text-align: justify">
      The TD0 is calculated as described before for example by Carl T. Wittwer (United States Patent US6303305B1). 
      In principle, the first and second derivative is calculated. To remove measurement noise, the second 
      derivative is smoothed by taking the mean of three adjacent cycles. The TD0 is called on as the zero 
      crossing in the third derivative of the smoothed second derivative. Note that calling of TD0 does not 
      require to define threshold and any baseline correction based on a linear function by the machine will 
      be lost in the second derivative.
    </p>
    <p style="text-align: justify">
      The calculation of the PCR efficiency is based on the inclination of the amplification curve when plotted 
      on a logarithmic fluorescence axis. RDML-RunAnalysis will first perform a baseline correction by 
      determining the exponential range, apply a baseline in the middle between 0 and the lowest 
      fluorescence, split the exponential range by half in a lower and an upper range and calculate their 
      inclinations. If the lower range has a lower inclination than the upper range, the baseline was 
      underestimated. Then a new baseline is applied as half of the value before and the lowest 
      fluorescence of the exponential range. If the inclination of the lower range was higher, the baseline 
      was overestimated and a new baseline is calculated as half of 0 and the value before. This continues 
      untill both inclinations differ less then 0.0001. From the upper inclination the individual PCR 
      efficiency is calculated. The PCR efficiencies have still some noise, so the user can choose to exclude 
      derivating efficiencies based on an statistical function, the difference to the median value or to 
      include all. The statistical function performs an iterative outlier test consisting of a combination of 
      skewness testing and, if a skewed distribution of E values is observed, testing for the presence of an 
      outlying value with a Grubbs test; if present, the outlier is excluded from the analysis and the test is 
      repeated. The remaining values are used to calculate a mean PCR efficiency for the target. Be aware 
      that this calculations do not work on data which were baseline corrected using a linear function with an 
      inclination.
    </p>
    <p style="text-align: justify">
      The TD0 and the PCR efficiency can be used to calculate an Ncopy value, which gives the copies present 
      in the reaction mix in the well.
    </p>
    <ol>
    <li>Open <a href="https://www.rdml-tools.com/runanalysis.html">RDML-RunAnalysis</a>, select the RDML file and press 
      "Upload File".</li>
    <li>Go to the "RunView" tab and select the desired experiment and run.</li>
    <li>Go to the "AmplificationCurveAnalysis" tab and click "Run AmplificationCurveAnalysis"</li>
    <li>A table is calculated with a row for each reaction with the data discussed before. Evaluate the excluded 
      and note messages.</li>
    <li>This table can be copied to the clipboard by clicking "Copy table to clipboard" and pasted in a spreadsheet 
      program or downloaded TSV file by clicking "Save table as CSV" for later use. </li>
    <li>A Double click on the row opens an individual curve in the "RunView" tab and allows the inspection of 
      the amplification curve. The thin black line shows the mean PCR efficiency a thin grey line the individual 
      PCR efficiency. The exponential phase is marked with big dots, if included in calculation, the dots are 
      filled.</li>
    <li>The curves should be inspected and reactions with issues should be excluded from further analysis.</li>
    <li>All reactions which are marked as excluded by any text in this column are ignored in further analysis. 
      On the RunView tab the excluded and notes field can be modified or deleted, provided a single reaction is 
      selected. Be aware that the table on the "Melting Curve Analysis" tab is not updated. Alternativley, this 
      can be done by clicking the link below "Edit a single Run in RunEdit" on the "Main" tab and following the 
      instructions of <a href="#RDML-RunEdit">RDML-RunEdit</a>.</li>
    <li>Save the analyzed RDML file by clicking "Save RDML" on the "AmplificationCurveAnalysis" tab.</li>
    </ol>





    <h5 id="RDML-Tools-Tips">2.x. Important Tips Using RDML-Tools</h5>
    <p style="text-align: justify">
      The Tools will discriminate between upper- and lower-case characters and handle them as different entries. 
      Technical replicate reactions are identified by having the same sample and target name. The sample type 
      allows the Tools to handle positive and negative controls correctly and the target type is required to 
      identify reference genes.
    </p>
    <p style="text-align: justify">
      On the "Main" tab of all RDML-Tools a "Links to other RDML tools" section is present which allows to open 
      the uploaded file in other modules. This allows to open the same file in different modules and work on 
      them at the same time. Before switching modules, reload the browser tab to update the data.
    </p>







    <h5 id="RDML-RunView">2.5. RDML-RunView</h5>
    <p style="text-align: justify">
      RDML-RunView is a tool to view single runs stored in a RDML files. Once the RDML file is loaded, the desired experiment
      and run can be selected. RDML-RunView will display the plate setup and, with classic qPCR, the amplification- or
      melt curves. A well / reaction can be selected, which will highlight the corresponding curve. If a curve is
      selected, the corresponding reaction will be highlighted. Melt curves can easily be inspected if the color coding
      is set to target. Setting the color coding to type allows easy inspection of amplification curves.
    </p>

    <h5 id="RDML-Validate">2.6. RDML-Validate</h5>
    <p style="text-align: justify">
      RDML-Validate validates a RDML file against the corresponding schema. If the validation fails, RDML-Validate
      displays a list of errors. Errors point to programming mistakes in the software used to create the RDML file and
      might be difficult to fix by the user. RDML-Edit might help to repaire files as described in 
      <a href="#Import-invalidRDMLfile">Import of a Invalid RDML file</a>. If this happens, you should contact the 
      developers of the respective software and send us the corrupted RDML file.
    </p>

    <h5 id="RDML-FixBrokenRDML">2.7. RDML-Fix Broken RDML Files</h5>
    <p style="text-align: justify">
      Some machines produce broken RDML files. RDML-Tools can fix some of these files:<br />
      1. Open the file in RDML-Edit.<br />
      2. "Enable Edit Mode" with the button on top.<br />
      3. Go to the Main tab.<br />
      4. Click the button "Recreate Lost IDs".<br />
      5. Click the button "Repair RDML File".<br />
      6. Save the repaired RDML file.<br />
      7. Click the link box below "Click here to validate RDML file".<br />
      8. If it does not state "RDML file is valid.", send us the RDML file for further inspection.
    </p>

 <ol>
    <li></li>
    <li></li>
    <li></li>
    <li></li>
    <li></li>
    <li>In case information is missing, there are several options. 
      |<ul style="list-style-type:disc;">
        <li>The information can be added in this tab using the "Complement columns 1 to 6" section. Single wells  
          like A7 or squares B2-D7 can be used. Sample and target may be any text, but be aware that technical 
          replicates must have identical sample and target information.</li>
        <li>The current file can be downloaded by pressing "Save table file in RDES format" on the top. Open 
          the file in Excel (Microsoft) or Calc (Libre Office) and complement the missing data and save the file. 
          It is recommended to import the file once more as described in <a href="#Import-repairRDESfile">Repair 
          a RDES file</a> to get the column seperators and line endings right.</li>
        <li>Lable all samples "sample1" and all targets "target1". Then the correct annotation must be provided 
          using RDML-Edit and other tools later. This could be more convenient.</li>
      </ul>
    <li>Now the file is ready. Save it by pressing "Save table file in RDES format" on the top.</li>
    <li>Repeat the procedure to have amplification and melting RDES files.</li>
    <li>Continue creating an RDML file following <a href="#Import-validRDESfile">Import of a Valid RDES 
      file</a></li>
    </ol>




    <h4 id="RDML-">3. RDML-RunAnalysis</h4>
    <ul>
      <li><a href="#LRP-Introduction">3.1. Introduction</a></li>
      <li><a href="#LRP-Warning">3.2. A Word of Caution</a></li>
      <li><a href="#LRP-References">3.3. LinRegPCR References</a></li>
      <li><a href="#LRP-Terms">3.4. Definitions of Terms Used in This Manual</a></li>
      <li><a href="#LRP-Theory-Behind">3.5. Theory Behind the LinRegPCR Program</a>
        <ul>
         <li><a href="#LRP-Basics">3.5.1. Basic Principle</a></li>
         <li><a href="#LRP-Monitoring-Chemistries">3.5.2. Cq and Efficiency with Different Monitoring Chemistries</a></li>
        </ul>
      </li>
      <li><a href="#LRP-Baseline">3.6. Baseline Estimation and Error Flagging</a>
        <ul>
          <li><a href="#LRP-Baseline-Basics">3.6.1 Baseline Estimation and Flagging of Deviating Reactions</a></li>
          <li><a href="#LRP-Error-Flagging">3.6.2. Error Flagging</a></li>
          <li><a href="#LRP-Failing-Baseline">3.6.3. Reasons for Failing Baseline Estimation and noisy samples</a></li>
          <li><a href="#LRP-Neg-Val-Warning">3.6.4. Negative Values Warning</a></li>
        </ul>
      </li>
      <li><a href="#LRP-Mean-Efficiency">3.7. Mean Efficiency Calculation</a></li>
      <li><a href="#LRP-WoL-Threshold">3.8. Window-of-Linearity and Quantification Threshold</a></li>
      <li><a href="#LRP-Diff-Pascal">3.9. Differences Between Pascal and Python Version</a></li>
      <li><a href="#LRP-Prepare-RDML">3.10. Importance of the information in RDML files</a></li>
      <li><a href="#LRP-Run">3.11. Run RunAnalysis</a></li>
    </ul>
    <h5 id="LRP-Introduction">3.1. Introduction</h5>
    <p style="text-align: justify">
      RunAnalysis is a program based on LinRegPCR for the analysis of real time PCR Data, also called quantitative PCR (qPCR) data. The
      program uses non-baseline corrected data, performs a baseline correction on each reaction separately, determines
      a window-of-linearity in the exponential phase and then uses linear regression analysis to determine the PCR
      efficiency per reaction from the slope of the regression line fitted to the selection of data points. The mean
      PCR efficiency per amplicon and the Cq value per reaction are used to calculate the starting concentration per
      reaction, expressed in arbitrary fluorescence units. Data input and output are in RDML format and annotated
      analysis results can be saved to a tab separated spreadsheet *.tsv files, which can be imported into programs
      like Excel (Microsoft) or Calc (OpenOffice).
      <br />
      The linear regression procedure used in this program can be found in every statistics textbook.
    </p>

    <h5 id="LRP-Warning">3.2. A Word of Caution</h5>
    <p style="text-align: justify">
      LinRegPCR is based on the procedures described in the paper by
      <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2665230/" target="_blank">Ruijter et al., Nucleic Acids
        Research, 2009 37(6):e45 (doi: 10.1093/nar/gkp045 or PMID: 19237396)</a>. The procedures in this program can
      be used for fluorescence data resulting from monitoring the PCR reaction. You should start your analysis with
      raw fluorescence data. Data import is your own responsibility. It is of importance to check which data are
      exported by your PCR system and how to retrieve the raw fluorescence data. As a quick and easy check; negative
      values should not occur in your raw fluorescent data file. The fact that LinRegPCR reads your data is <b>no
      proof</b> that you have the correct data. By using this program you acknowledge that you have read the above
      paper, understand it, and agree with its conclusions. Therefore, you assume all responsibility and liability
      for the selection of this program to achieve your intended goals, and for the conclusions you draw from the
      analysis results. The developers of this application cannot be held responsible for any consequences of the
      use of this program. Please read the
      <a class="plain" target="_blank" href="https://www.gear-genomics.com/terms">Terms of Use</a>
      before using LinRegPCR and please obey to the requirements.
    </p>

    <h5 id="LRP-References">3.3. LinRegPCR References</h5>
    <p style="text-align: justify">
      When you refer to LinRegPCR in your paper, we prefer that you use the reference
      <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2665230/" target="_blank">Ruijter et al., Nucleic Acids
        Research, 2009 37(6):e45 (doi: 10.1093/nar/gkp045 or PMID: 19237396)</a>.
      When you are using hydrolysis probes, you can
      also use the reference Tuomi et al. (2010) Methods 50: 313–322. The handling of other chemistries is described
      in Ruijter et al. (2014) Microchimica Acta, 181(13-14):1689-1696). A comparison of the different amplification
      curve analysis approaches can be found in Ruijter et al. (2013) Methods 59: 32-46.
      <br /><br />
      <i>Please do not use the abbreviation LinReg when you refer to LinRegPCR.</i>
      <br /><br />
      Ruijter JM, Lorenz P, Tuomi JM, Hecker M, van den Hoff MJB (2014)<br /><b>Fluorescent increase kinetics of
      different fluorescent reporters used for qPCR depends on monitoring chemistry, targeted sequence, type of
      DNA input and PCR efficiency.</b><br />
      <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4167442/" target="_blank">Microchimica Acta.
      181(13-14):1689-1696.</a><br /><i>This paper is an extension of Tuomi et al. (2010) and describes the processing
      of qPCR data resulting from 6 different categories of monitoring chemistries.</i>
      <br /><br />
      Ruijter JM, Pfaffl MW, Zhao S, Spiess AN, Boggy G, Blom J, Rutledge RG, Sisti D, Lievens A, De Preter K,
      Derveaux S, Hellemans J, Vandesompele J. (2013)<br /><b>Evaluation of qPCR curve analysis methods for reliable
      biomarker discovery: bias, resolution, precision, and implications.</b><br />
      <a href="https://www.ncbi.nlm.nih.gov/pubmed/22975077" target="_blank">Methods 59: 32-46.</a>
      <br /><i>This paper describes the comparison of the performance, in terms of precision, linearity, and
      sensitivity of all available qPCR curve analysis methods using the same biomarker data sets.</i>
      <br /><br />
      Tuomi JM, Voorbraak F, Jones DL, Ruijter JM. (2010)<br /><b>Bias in the Cq value observed with hydrolysis probe
      based quantitative PCR can be corrected with the estimated PCR efficiency value.</b><br />
      <a href="https://www.ncbi.nlm.nih.gov/pubmed/20138998" target="_blank">Methods 50, 313–322.</a>
      <br /><i>This paper describes the use of LinRegPCR in the analysis of cumulative fluorescence datasets obtained
      with hydrolysis probes like Taqman probes.</i>
      <br /><br />
      Ruijter JM, Ramakers C, Hoogaars W, Bakker O, van den Hoff MJB, Karlen Y, Moorman AFM. (2009)<br /><b>Amplification
      efficiency: linking baseline and bias in the analysis of quantitative PCR data.</b><br />
      <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2665230/" target="_blank">Nucleic Acids Research 37: e45.</a>
      <br /><i>This paper describes the basics of the LinRegPCR program and gives details on the fluorescence baseline
        correction and setting of the window-of-linearity.</i>
      <br /><br />
      Ramakers C, Ruijter JM, Lekanne Deprez RH, Moorman AFM. (2003)<br /><b>Assumption-free analysis of quantitative
      real-time PCR data.</b><br />
      <a href="https://www.ncbi.nlm.nih.gov/pubmed/12618301" target="_blank">Neurosci Letters 339: 62-66.</a>
      <br /><i>This paper describes the estimation of PCR efficiencies for individual reactions from a subset of data
      points in the log-linear phase of the amplification curves. The use of individual efficiencies to calculate a
      target quantity per reaction is no longer recommended (see papers: Ruijter et al., 2009 and Ruijter et al.,
      2013).</i>
    </p>

    <h5 id="LRP-Terms">3.4. Definitions of Terms Used in This Manual</h5>
    <p style="text-align: justify">
      <b>Amplicon group:</b> A set of reactions in which the same pair of primers is used to amplify the target or
      DNA-of-interest (=amplicon).
      <br /><br />
      <b>Baseline:</b> Measured fluorescence when no amplification-specific fluorescence can be determined yet. This
      baseline fluorescence includes fluorescence from cDNA, primers and unbound SYBR Green, inefficiently quenched
      fluorescently labeled probe and system noise.
      <br /><br />
      <b>Cq or quantification cycle:</b> Fractional number of cycles needed to reach the fluorescence threshold.
      Reactions with higher starting concentrations will reach this threshold earlier and will have a low Cq value;
      Cq is used to calculate the target quantity (N0). Cq is also known as Cp or Ct but the use of those terms is
      discouraged by the MIQE guidelines (Bustin et al., 2009 Clinical Chemistry 55:611-622)
      <br /><br />
      <b>Cumulative fluorescence:</b> Observed fluorescence data obtained when the PCR amplification is monitored with
      a monitoring system for which the fluorophore is released in the reaction mix and thus accumulates during
      subsequent cycles.
      <br /><br />
      <b>E or PCR Efficiency:</b> Amplification efficiency is defined as "the fold increase per cycle", and is thus a
      value between 1 and 2. This efficiency is calculated from the slope of the exponential phase of the amplification
      curve plotted on a logarithmic fluorescence axis (Fig. 1). Ideally the PCR efficiency is 2, meaning that in each
      cycle the amount of amplicon doubles. This is also referred to as a 100% efficient PCR. An efficiency value of
      1 means no amplification. The mean PCR efficiency calculated for the different reactions per amplicon is used
      to calculate the target quantity per reaction (see Eq. 2).
      <br /><br />
      <b>Exponential or log-linear phase:</b> The portion of the PCR curve in which the level of generated fluorescence
      exceeds baseline fluorescence and reagents are not yet limiting. In this phase the amplification efficiency is
      constant between cycles and similar across all reactions in an amplicon group regardless of their starting
      concentration. The exponential phase can be identified as the data points on a straight line when the
      amplification curve is plotted on a logarithmic fluorescence axis.
      <br /><br />
      <b>Fluorescence threshold, quantification threshold or Nq:</b> Fixed amount of fluorescence, set to determine the
      Cq value. By default, this web version of LinRegPCR sets a common threshold for all amplicons (see below).
      <br /><br />
      <b>N<sub>0</sub>:</b> Pronounced N-zero. Target quantity or starting concentration per reaction, expressed in
      arbitrary fluorescence units and calculated with Eq. 2 (see below). The N<sub>0</sub> value of a target and one
      or more reference genes can be used to calculate the efficiency-corrected gene expression ratio.
      <br /><br />
      <b>N<sub>copy</sub>:</b> Is calculated as N<sub>copy</sub> = 10 * 1.9^35 * N<sub>0</sub> / Fluorescence threshold<br> 
      N<sub>copy</sub> is applying the "rule of thumb" indicating that a sample with 10 copies reaches after 35 cycles 
      the threshold. Of course this is rarely true. It is based on the assumption, that the threshold level is allways 
      corresponding to the same DNA concentration. The length of the amplicon, the volume of the reaction, the 
      influence of the chemisty and fluorochrome concentration are ignored in this calculation. Nevertheless this values 
      are independent of the machine fluorescence levels and provide a human understandable copy number (for the reaction 
      tube). This values should not be more than factor 10x off.
      <br /><br />
      <b>Non-cumulative fluorescence:</b> Observed fluorescence data obtained when the PCR amplification is monitored
      with a DNA binding dye or hybridization probe. The reporter is only fluorescent when bound to dsDNA; fluorescence
      intensity is proportional to the amount of DNA present at each cycle.
      <br /><br />
      <b>Nq:</b> see Fluorescence threshold.
      <br /><br />
      <b>Plateau phase:</b> Last phase of the PCR reaction - the phase in which reaction components become limiting. In
      the plateau phase the amplification curves of the real-time PCR are no longer exponential and the PCR efficiency
      drops to 1.
      <br /><br />
      <b>Raw data:</b> Fluorescence data that is not baseline-corrected by the PCR system. Raw data should not contain
      negative values.
      <br /><br />
      <b>Reaction:</b> The qPCR (well in a PCR plate) from which fluorescence data are recorded.
      <br /><br />
      <b>Tissue or Sample:</b> Biological material in which DNA (qPCR) or RNA (RT-qPCR) levels are determined. A tissue
      can be divided in several reactions (wells) in which the same or different targets can be measured.
      <br /><br />
      <b>Window-of-Linearity or W-o-L:</b> Subset of data points in the exponential phase that is used to determine the
      PCR amplification efficiency (E) per reaction. The Window-of-Linearity is set automatically by the program.
      Default the W-o-L width is set to include 4 amplification cycles for each reaction.
    </p>

    <h5 id="LRP-Theory-Behind">3.5. Theory Behind the LinRegPCR Program</h5>
    <h5 id="LRP-Basics">3.5.1. Basic Principle</h5>
    <p style="text-align: justify">
      <b>Basic principle</b><br />
      The basic equation for PCR kinetics is:<br />
      <img src="static/img/LinRegPCR_Formula_1.jpg" alt="Formula" width="250px"><br />
      Where the amount of amplicon after C cycles (N<sub>c</sub>) is the starting concentration of the amplicon
      (N<sub>0</sub>) times the PCR efficiency (E) to the power the number of cycles (c). The PCR efficiency in this
      equation is defined as the fold increase per cycle.
      <br /><br /><br /><img src="static/img/LinRegPCR_Fig1.jpg" alt="Formula" width="750px"><br /><br /><br />
      LinRegPCR is based on the fact that, when plotted on a log-fluorescence axis, the slope of the linear regression
      line through the data points in the exponential phase of the amplification curve is log(E) and can thus be used
      to estimate the PCR efficiency (E) (Ramakers et al 2003).
      <br />
      After setting the quantification threshold Nq and determining the fractional cycle number that was needed to
      reach this threshold (Cq, Fig. 1), the starting concentration or target quantity can be calculated with the
      inverse of Eq. 1:<br />
      <img src="static/img/LinRegPCR_Formula_2.jpg" alt="Formula" width="250"><br />
      The Efficiency value used in Eq. 2 is the mean of the observed efficiencies from the different reactions in which
      the same target is amplified (Ruijter et al 2009). Note that the use of the efficiency per amplicon in the
      calculation of N<sub>0</sub> means that this result is the efficiency-corrected target quantity.
    </p>

    <h5 id="LRP-Monitoring-Chemistries">3.5.2. Cq and Efficiency with Different Monitoring Chemistries</h5>
    <p style="text-align: justify">
      In PCR runs monitored with a DNA binding dye, the fluorescence is linearly related to the amount of the amplicon.
      Eq. 1, therefor, describes the increase in fluorescence. However, for some monitoring chemistries, the fluorescent
      reporter is released from quenching during elongation and remains fluorescent in subsequent cycles; the
      fluorescence thus accumulates. For other chemistries, the first release of fluorescence occurs only after a
      number of initial lag-cycles. Some chemistries combine a lag with accumulation. Plotted on a logarithmic
      fluorescence axis every chemistry shows the amplification curve for each reaction as a straight line, with a
      slope equal to log(E). This is true for all chemistries used for monitoring DNA increase in a real-time PCR run
      (Ruijter et al 2014).
      <br />
      However, the different kinetics lead to a shift in the Cq values of these curves and the correction for the
      observed, earlier or later, Cq values are implemented in LinRegPCR (Tuomi et al 2010; Ruijter et al., 2014). The
      required correction depends on the chosen monitoring chemistry and the input DNA (ss cDNA or ds DNA); the
      targeted sequence in case of fluorescent probes (targeting the mRNA or cDNA strand) does not displace the
      amplification curve observably.
      <br />
      Both the PCR efficiency estimation and the C<sub>shift</sub> correction for cumulative chemistries are only valid
      when the Cq value of the reaction is not too low. Therefore, the program checks the Cq value against the critical
      values. When the Cq of a reaction is too low to allow a valid PCR efficiency to be estimated, the reaction is
      excluded from the calculation of the mean efficiency. LinRegPCR will use an accepted error in efficiency of 0.01
      (=1%) and an accepted bias in N<sub>0</sub> of 0.05 (=5%). The Cq value can be too low to perform the right
      C<sub>shift</sub> correction which will affect N<sub>0</sub>. In that case the program will flag the reaction in
      the results in the respective column.
    </p>

    <h5 id="LRP-Baseline">3.6. Baseline Estimation and Error Flagging</h5>
    <h5 id="LRP-Baseline-Basics">3.6.1 Baseline Estimation and Flagging of Deviating Reactions</h5>
    <p style="text-align: justify">
      LinRegPCR determines the baseline fluorescence per reaction by reconstructing the log-linear phase. The program
      determines the cycle at the maximum of the Second Derivative (SDM) of the observed fluorescence values in each
      reaction. The cycles after the SDM cycle are considered the transition into the plateau phase, referred to as
      the transition phase. Then LinRegPCR does an iterative search for a baseline value that gives the largest number
      of data points on a straight line down from the SDM cycle. For details on the baseline estimation procedure: see
      Ruijter et al., 2009. As explained in that paper, an accurate baseline estimation is a requirement for the
      correct determination of the PCR efficiency and to a lesser extend the correct Cq value. The slope of the
      amplification curve depends on the subtracted baseline value (Fig. 2 C, D and E). The implemented baseline
      estimation requires a sufficiently long log-linear phase (at least 4 points) and thus optimized qPCR conditions
      must be used. When the baseline-to-plateau distance is too small the baseline determination can fail. LinRegPCR
      determines and flags the following deviating reactions during the process of baseline estimation.
    </p>

    <h5 id="LRP-Error-Flagging">3.6.2. Error Flagging</h5>
    <p style="text-align: justify">
      <b>no amplification:</b> reactions with less than 7-times overall increase in fluorescence between the first
      cycles and the last cycles. These reactions will be ignored by the program and are always excluded from the
      calculation of the mean PCR efficiency. No starting concentration will be calculated for these reactions.
      <br /><br />
      <b>baseline error:</b> reactions for which no baseline value can be determined because no fluorescence value
      between the SDM and the minimum observed value results in a straight exponential phase. These reactions will
      be ignored by the program and are always excluded from the calculation of the mean PCR efficiency. No starting
      concentration will be calculated for these reactions.
      <br /><br />
      <b>noisy sample:</b> reactions in which the fluorescence in cycle C is higher than it is in cycle C+1 in a
      W-o-L (see Fig. 2A). This should not happen because in the exponential phase, the fluorescence values should be
      continuously increasing. When it happens, the baseline estimation will result in a warning that the dataset is
      <b>too noisy</b> to be analyzed automatically.
      <br /><br />
      <b>no-plateau:</b> reactions that do not reach the plateau phase. The user can decide to include or exclude
      the <b>no-plateau</b> reactions from the calculation of the mean PCR efficiency. For these reactions a starting
      concentration will be calculated when they are not an <b>efficiency outlier</b>.
    </p>

    <h5 id="LRP-Failing-Baseline">3.6.3. Reasons for Failing Baseline Estimation and noisy samples</h5>
    <p style="text-align: justify">
      When you encounter a lot of baseline errors and/or noisy reactions it may be that the Log-linear phase in your
      dataset is too short. This is most probably because the baseline is too high which is limiting the
      baseline-to-plateau distance. The program then has not enough data points in the log-linear phase to estimate
      the fluorescence baseline. It may also be that measurement noise leads to discontinuous data in the log-linear
      phase. Both problems make that the data points downwards from the plateau phase do not form a continuously
      decreasing set of points. Because LinRegPCR uses only the points that are continuously decreasing from the
      plateau phase not enough datapoints are available for correct baseline estimation. In that case the baseline
      estimate will be wrong, in most cases too high.
      <br /><br /><br /><img src="static/img/LinRegPCR_Fig2.jpg" alt="Formula" width="750px"><br /><br /><br />
      To solve this issue, you have to go back to the lab. Firstly, look at the exposure-time setting in your PCR
      apparatus and set this to a fixed exposure time. The automatic exposure time setting of your qPCR apparatus
      exaggerates noise at low fluorescence intensities. Secondly, you can try to increase the baseline - plateau
      distance by lowering the primer concentration in the PCR reaction. When you are using fluorescent probes and
      quenched reporters you have to make sure that the quenching efficiency of the probe is as high as possible (see
      Tuomi et al 2010 for suggestions).
    </p>

    <h5 id="LRP-Neg-Val-Warning">3.6.4. Negative Values Warning</h5>
   

    <h5 id="LRP-Mean-Efficiency">3.7. Mean Efficiency Calculation</h5>
    <p style="text-align: justify">
      The group of reactions, in which the same target (pair of primers) is used, is called an amplicon group. The PCR
      efficiency per amplicon is assumed to be constant; nevertheless, individual reactions can have slightly variable
      observed PCR efficiencies. Therefore, it is recommended to use the average efficiency of the reactions for each
      amplicon group to calculate the target quantity per reaction (Ruijter et al., 2009). Determining the correct mean
      efficiency is very important and deviating reactions should, therefore, be excluded. However, in some cases the
      user can decide to include deviating reactions in the calculation of the mean efficiency.
      <br /><br />
      <b>No amplification:</b> Reactions without amplification are always excluded.
      <br /><br />
      <b>No Plateau:</b> Reactions that do not reach the plateau phase can be included by the user when the number of
      reactions on which the mean efficiency is based becomes too low. However, excluding these reactions may give
      you a better mean efficiency.
      <br /><br />
      <b>Cq too low:</b> Reactions of which the Cq value is too low to determine the correct PCR efficiency (only with
      cumulative fluorescence) are excluded. Excluded reactions are only excluded from the calculation of the mean
      efficiency per amplicon group. The reaction will receive a starting concentration (N<sub>0</sub>) in the output
      based on the mean PCR efficiency of the remaining reactions. This is because the Cq value of these reactions,
      when observed, is probably correct.
      <br /><br />
      <b>Cq too high (only warning):</b> Reactions that do not reach the plateau and have a Cq value above 35. Such a
      high Cq value means that the starting concentration is 10 copies or less which makes that the Poisson effect has
      large influence and the calculated target quantity is unreliable (Nolan et al., Nature Protocols, 2006). The user
      is advised to replace the Cq value with the maximum Cq value observed for the target with one copy per reaction
      (De ronde et al., Nucl Acids Res, 2017).
      <br /><br />
      <b>Efficiency outliers:</b> Reactions with a deviating individual PCR efficiency are excluded but can be included
      in the calculation of the mean efficiency. The default, and preferred, way to determine efficiency outliers is
      the use of the combined skewness and Grubb’s test for outliers causing a skewed distribution of PCR efficiency
      values.
      <br /><br />
      Alternatively, you can give the range around the median efficiency that you want to allow. The default range of
      5% means that, for instance, around a median efficiency of 1.83, a range from 1.78 to 1.88 is allowed. This
      "exclusion range" option was the default in the Windows version of the program and is, therefore, preserved in
      this web version of LinRegPCR.
      <br /><br />
      For efficiency-outlier reactions no target quantity will be calculated because the deviating efficiency
      will also have biased the observed Cq value.
    </p>
    <h5 id="LRP-WoL-Threshold">3.8. Window-of-Linearity and Quantification Threshold</h5>
    <p style="text-align: justify">
      To determine the mean PCR efficiency for an amplicon group, LinRegPCR uses a selection of data points in the
      log-linear phase to determine the PCR efficiency (E) per reaction. This selection of cycles is called the
      <b>Window-of-Linearity (W-o-L)</b>. After the baseline correction, LinRegPCR sets a W-o-L per amplicon group by
      looking for the W-o-L with the lowest coefficient of variation (CV = SD / mean) of the individual PCR
      efficiencies in the group of reactions. To this end, LinRegPCR moves a W-o-L, starting at the plateau phase,
      downward till reactions are no longer included or till the CV shows a second minimum. The latter criterion was
      added after the publication of the method (Ruijter et al., 2009) to avoid the situation where the increasing
      average efficiency close to the baseline causes the CV to decrease. This procedure ensures that the W-o-L is set
      in such a way that the variation between individual efficiency values of the reactions in the group is lowest. The
      W-o-L is, by default, set to include 4 data points for each reaction.
      <br />
      After setting the W-o-L, a common quantification threshold is automatically set at the geometric mean of 1 cycle
      below the upper limit of the W-o-L’s of the different amplicon groups in the run.
      <br />
      When an individual reaction or a target is selected within the RunView Tab in the LinRegPCR app, the W-o-L is
      indicated by blue horizontal lines that represent its upper and lower limit. The data points used for the
      estimation of the PCR efficiency are solid dots. The quantification threshold is represented by the horizontal
      green line; the vertical green line indicates the Cq value; the thin black line indicates the mean PCR efficiency;
      the thin grey line indicates the individual PCR efficiency.
    </p>
    <h5 id="LRP-Diff-Pascal">3.9. Differences Between Pascal and Python Version</h5>
    <p style="text-align: justify">
      LinRegPCR was originally written in Pascal/Delphi as a standalone software for the Windows platform. It was
      entirely reprogrammed using Python and Numpy library and is provided as a web application that can be used from
      any computer with internet access and a modern web browser. Bioinformaticians should use the
      <a href="https://github.com/RDML-consortium/rdmlpython" target="_blank">python library</a> directly to avoid too
      much load on the server.
      <br />
      The Pascal and Python version have in common, that with the same input and user parameters, both will calculate
      exactly the same result. The result itself is presented in a different column format and while the old Pascal
      version gives -999 as a pre-defined error value, the Python version gives no result when no target quantity is
      calculated. If you find different results when you re-do an earlier LinRegPCR analysis with the new web
      version, please contact us.
      <br />
      The original Pascal version of the program is reading the raw fluorescence data from different machine export
      formats. The new Python version works solely on RDML files, which need to be well formatted (see section 2).
      The first step is to use RDES-TableShaper and RDML-Edit to import the raw fluorescence data, that were exported
      by the qPCR machine, into a new RDML file. Spend some time to get the sample ids and reaction types and the
      target information right, as LinRegPCR reads them and bases its analysis on those data.
      <br />
      The original Pascal version evolved iteratively over almost two decades. It offers many user options and the
      manual setting of several obsolete parameters. The new Python version only offers automated and no manual
      analysis. This choice has the advantage that the analysis time is down from minutes to seconds. The calculations
      are always per target and reaction (there is no common calculation over all reactions) but the Cq values are
      reported for a common quantification threshold. The latter choice makes that Cq values can be directly compared.
      Note, however that the aim of a qPCR experiment is not the Cq value but the efficiency-corrected target quantity,
      and/or the gene expression ratio derived from this quantity for different targets.
    </p>
    <h5 id="LRP-Prepare-RDML">3.10. Importance of the information in RDML files</h5>
    <p style="text-align: justify">
      The subset of reactions, in which the same target (pair of primers) is used, is called an amplicon group.
      Individual reactions can have slightly variable observed PCR efficiencies but the PCR efficiency per amplicon is
      assumed to be constant. Therefore, LinRegPCR uses the mean PCR efficiency per amplicon in its calculations
      (Ruijter et al., 2009). To enable this, the reactions in the run have to be assigned to amplicon groups.
      <br />
      The estimation of the mean PCR efficiency per target depends on a good baseline correction which can only be
      performed on raw fluorescence data. . As there are many formats for export of raw fluorescence data by qPCR
      machines, the data need to be reshaped to the RDML import format. RDES-TableShaper and RDML-Edit will assist
      you with this task.
      <br />
      Because LinRegPCR calculates a mean PCR efficiency per target, it is essential, that the sample id (= sample
      name) and the target are correctly annotated (optimally in RDES-TableShaper). To decide whether the analysis
      result is correct or not, the program needs to ‘know’ which reactions are supposed to show amplification and
      which are not. For instance, no_amplification is what you expect in a negative control and not in a positive
      control. In the RDML file each reaction should have the correct sample type, because the warnings will depend
      on the reaction type and be different for unknown, positive or negative reactions.
      <br />
      Different monitoring chemistries require different Cq corrections which makes it of importance to use RDML-Edit
      to update the respective dye element.
    </p>
    <h5 id="LRP-Run">3.11. Run RunAnalysis</h5>
    <p style="text-align: justify">
            LinRegPCR depends on correctly annotated RDML files, see "How to Prepare RDML files" and section 2 for details.
      As described before, the LinRegPCR program is used for the calculation of the starting concentration
      (N<sub>0</sub>) with Eq. 2, using Nq (the quantification threshold), Cq (the fractional number of cycles that
      was needed to reach this threshold), and E (the mean PCR Efficiency per amplicon; see Eq 2). Therefore, it is
      essential that Cq and E are determined as accurate as possible. To this end LinRegPCR imports non-baseline
      corrected data, performs a baseline correction on each reaction separately, determines a window-of-linearity
      per amplicon and then uses linear regression analysis to fit a straight line to the selected PCR data per
      reaction. Subsequently an individual E is calculated from the slope of this regression line and, per target,
      the mean efficiency per amplicon is calculated. A common Nq is set and Cq is determined per reaction. Then the
      N<sub>q</sub>, the mean PCR efficiency per amplicon and the Cq per reaction are used to calculate the starting
      concentration per reaction (Eq. 2). This N<sub>0</sub> is expressed in arbitrary fluorescence units but when the
      relative gene expression is determined the fluorescence units are cancelled out.
      <br />
      In the LinRegPCR interface the tabs have to be worked at from left to right. If information or choices on a tab
      is modified, all information in the later (more to the right) tabs is overwritten and must be re-entered.
      Selection of reactions for highlighting in the RunView tab or by double-clicking on the results table do not
      change entered data or choices. Modifications of fields requiring a new run of LinRegPCR have a grey background.
      <br />
      The basic steps of the use of the program are:
      <br /><br />
      1) Upload the file on the main (most left) tab by selecting the file and clicking the "Upload File" button.
      <br /><br />
      2) Select the desired Experiment and Run on the RunView tab. Each change in these fields will delete or
      invalidate calculations of LinRegPCR. The other choices on the RunView tab will only influence the display of
      the curves below and can be adapted without influencing the calculations.
      <br /><br />
      3) The main parameters are set on the LinRegPCR tab. As baseline correction is a critical step, you have to
      upload raw data which are not baseline corrected and then leave the baseline correction to LinRegPCR. Secondly
      you can select the acceptable range for the PCR efficiency. Reactions outside this range will be flagged and
      can be excluded from the mean calculation. Preferably you should use the combined skewness and Grubb’s test to
      statistically identify efficiency outliers. By default, efficiency outliers are excluded from further
      calculations.
      <br /><br />
      4) Clicking on the "Run LinRegPCR" button will now calculate the result table. The exact results will depend on
      the choices about which reactions to exclude from mean efficiency calculation. The RDML file can only hold one
      result. Therefore, you must decide when you want to save the result to RDML. If "Update RDML Data" is yes, the
      results are written in the uploaded RDML file and earlier analysis results will be overwritten. You may choose to
      save it with a new filename. If "Update RDML Data" is no, the RDML file is not modified, but the result table is
      calculated. Note that each change requires a new calculation. <br />
      <b>A fresh calculation with LinRegPCR is run by clicking the "Run LinRegPCR" button.</b>
      <br /><br />
      5) LinRegPCR calculates exclusion and note strings. If a single reaction is selected by double-clicking on the
      results table in the LinRegPCR tab, the corresponding fluorescence data are displayed in the RunView tab and
      the exclusion and note strings are loaded below the curves, where they can be edited and saved back into the
      RDML file. Note that when the "Run LinRegPCR" button is hit again, the new calculation will overwrite the
      edited notes.
      <br /><br />
      6) Save your modified RDML file with the buttons on the Main tab or those on the LinRegPCR tab. The analysis
      results table can be saved with the button on the LinRegPCR tab and loaded into spreadsheet programs, like
      Excel or Calc. Do not forget to select the correct decimal separator for your region.
    </p>


    <h4 id="RDML-ExperimentAnalysis">4. RDML-ExperimentAnalysis</h4>
        <ul>
          <li><a href="#ANA-Introduction">4.1. Introduction</a></li>
          <li><a href="#ANA-Plate-Correct">4.2. InterRunCorrection</a></li>
          <li><a href="#ANA-Absolute-Quan">4.3. AbsoluteQuantification</a></li>
        </ul>
    <h5 id="ANA-Introduction">4.1. Introduction</h5>
    <p style="text-align: justify">
      RDML-ExperimentAnalysis is a program for the analysis of an qPCR experiment after N0 values have been calculated. The N0
      values can be calculated for each run using RDML-LinRegPCR.
    </p>

    <h5 id="ANA-Plate-Correct">4.2. InterRunCorrection</h5>
    <p style="text-align: justify">
      InterRunCorrection is based on the procedures described in the paper by
      <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4822202/" target="_blank">Ruijter et al., Removal of
      between-run variation in a multi-plate qPCR experiment. Biomol Detect Quantif. 2015 Jul 30;5:10-4. (doi:
      10.1016/j.bdq.2015.07.001 or PMID: 27077038)</a>. The procedures in this program can to calculate a correction
      factor for each run of an experiment and to use this factor to calculate corrected N0 and Cq values.
      <br /><br />
      The calculation of the correction factor is performed in several steps. First, the correction factors for all
      possible combinations of runs are calculated, comparing two runs at a time. This comparison is based on
      overlapping conditions which have to be present on both runs and may be modified with the selection "Overlap":
    </p>
    <ul style="list-style-type: circle">
      <li>
        If "Technical Replicates" is selected, overlapping conditions have to have identical target id and sample id.
        This is the default choice and equivalent to select technical replicates.
      </li>
      <li>
        If "Same Annotation" is selected, overlapping conditions have to have identical target id and the sample has to
        have an identical annotation key with an identical value. The annotation key to use can be selected with the
        selection "Sel. Annotation". Annotations can be used to select biological replicates instead of technical
        replicates (Annotations are set and edited using RDML-Edit at the sample tab).
      </li>
    </ul>
    <p style="text-align: justify">
      Identical overlapping conditions are used to calculate the difference between the runs comparing all possible
      combinations. For example, if on run 1 three technical replicates (A, B and C) and on run 2 two technical
      replicates (a, b) of the same reaction mix would be run, the possible differences A/a, A/b, B/a, B/b, C/a ans
      C/b would be calculated. This is done for all overlapping conditions. The difference between the two runs
      is the geometric mean of all the possible differences. The geometric mean is calculated separately for each
      target and for each run (combining all targets). The individual correction factor from the two runs
      are used to find the optimal correction factors for each run in the experiment. The selection "Correction Level"
      chooses if one combined factor is used per run ("run") or individual factors per target ("target").
      <br /><br />
      In the output the first table shows the run correction factors for each run.
      The second table shows for each target the individual and combined PCR efficiency and combined PCR
      efficiency standard error.
      The third table shows the combined threshold followed by the threshold per run.
      The last table provides a matrix with the number of overlapping conditions per target. This view could be
      helpful if an additional run must be planned to improve the overlap between the runs.
      <br /><br />
      The corrected N0 and Cq values are available on the RunView tab and are ideally viewed as data view "list".
      The table can be saved as tsv file. The RDML-File can be updated and save the calculation results for later
      use.
      <br /><br />
      By using this program you acknowledge that you have read the above
      paper, understand it, and agree with its conclusions. Therefore, you assume all responsibility and liability
      for the selection of this program to achieve your intended goals, and for the conclusions you draw from the
      analysis results. The developers of this application cannot be held responsible for any consequences of the
      use of this program. Please read the
      <a class="plain" target="_blank" href="https://www.gear-genomics.com/terms">Terms of Use</a>
      before using RDML-ExperimentAnalysis and please obey to the requirements.
    </p>

    <h5 id="ANA-Absolute-Quan">4.3. AbsoluteQuantification</h5>
    <p style="text-align: justify">
      The calculated N0 values are absolute values and can be used in calculations like concentrations. N0 gives the
      calculated fluorescence at the start of the qPCR run and depends therefore on the fluorescence values measured
      by the machine. The AbsoluteQuantification tools can convert the N0 values in more intuitive units like copies
      per microliter.
      <br /><br />
      The algorithm for this conversion can be chosen with the selection "Method":
    </p>
    <ul style="list-style-type: circle">
      <li>
        The "reference" approach requires samples with known concentration. The samples must be of the type "std"
        (standard sample) and must have the expected quantity like 1000 cop (copies per microliter).
        AbsoluteQuantification searches for the highest concentration to avoid dilution errors. If more than one
        sample of the highest concentration are present, the geometric mean of these samples is calculated and used
        to convert the N0 values.<br />
        Be aware, that one sample with a high concentration is sufficient, dilutions are not required for the
        calculation. If provided, the dilution error is calculated and the additional dilutions are ignored.
        This algorithm may be used provide the concentration relative to a reference sample which should be set to
        one. The reference sample is set in RDML-Edit to the type "std" and the quantity 1 fold.
      </li>
      <li>
        The "Rule of Thumb (Cq 35 == 10 copies)" approach is based on the common word of mouth saying 10 copies at
        start result in a Cq of 35. It was extended by an PCR efficiency of 1.9, a reaction volume of 20 &micro;l
        and a amplicon length of 100 bp. With the threshold and N0 as only input the copy number can be estimated.
        <br />
        Be aware, that this is a rough estimation ignoring different machine noise, differences between chemistries
        and contaminations. This rule would expect for the same sample the same Cq with any chemicals and on any
        machine. The approach is nice as it works on all qPCR data, but ist should only be used to get a rough
        estimation at which range the copy number is.
      </li>
      <li>
        The "Optical Calibrator" approach is experimental and works by providing a qPCR reaction with a final
        concentration of 10 ng/&micro;l PCR product and without primers and a second qPCR reaction without product
        and without primers. Both samples need to be of the type "opt" (optical calibrator) and must have the
        expected quantity like 10 ng. In the first five cycles the fluorescence of this product is used to measure
        DNA concentration without any PCR amplification, only based on binding of the fluorochrome. The fluorescence
        a concentration of 10 ng/&micro;l provides is used to calculate the concentration at a given N0.<br />
        copies = [conc in ng] * Mol / ([amplicon length in bp] * 660 * 1e9)<br />
        Be aware, that this is an experimental feature. It requires that qPCR machines do not measure the starting
        fluorescence and use it to correct the raw values per reaction. First experiments have an error by the
        factor 10.
      </li>
    </ul>
    <p style="text-align: justify">
      The first table provides an overview of the parameters at threshold for each target.
      The second table shows for each target the individual and combined PCR efficiency and combined PCR
      efficiency standard error.
      The third table shows the combined threshold followed by the threshold per run.
      The last table shows the standard samples with the expected and calculated concentrations for comparison.
      <br /><br />
      The final concentrations are available on the RunView tab and are ideally viewed as data view "list".
      The table can be saved as tsv file. The RDML-File can be updated and save the calculation results for later
      use.
      <br /><br />
      By using this program you acknowledge that you have read the above
      text, understand it, and agree with its conclusions. Therefore, you assume all responsibility and liability
      for the selection of this program to achieve your intended goals, and for the conclusions you draw from the
      analysis results. The developers of this application cannot be held responsible for any consequences of the
      use of this program. Please read the
      <a class="plain" target="_blank" href="https://www.gear-genomics.com/terms">Terms of Use</a>
      before using RDML-ExperimentAnalysis and please obey to the requirements.
    </p>


    <h4 id="RDML-qPCR-Basics">5. qPCR Basics</h4>
        <ul>
          <li><a href="#BASE-Introduction">5.1. Introduction</a></li>
          <li><a href="#BASE-qPCR-Setup">5.2. Setting up a qPCR</a></li>
          <li><a href="#BASE-Genomic-Std">5.3. Use genomic DNA as Standard</a></li>
        </ul>
    <h5 id="BASE-Introduction">5.1. Introduction</h5>
    <p style="text-align: justify">
      This section explains how to set up a qPCR reaction. Please consider it as one example, the way we do it. It 
      should provide guidance if required, but we do not promote it as the "only correct way". The products used 
      in this section are the ones we use; we do not wish to advice their use above other products. This section 
      would be the answer if you ask A. Untergasser: How would you do qPCR?
    </p>

    <h5 id="BASE-qPCR-Setup">5.2. Setting up a qPCR</h5>
    <p style="text-align: justify">
      qPCR requires good primers. Pick required primers using 
      <a href="https://www.primer3plus.com/" target="_blank">Primer3Plus</a> and order them.
    </p>
    <p style="text-align: justify">
      Upon arrival of the primers dissolve them in TE (10mM Tris, 1mM EDTA, pH 8.0) to a stock concentration of 
      100&micro;M (100&micro;mol/L). In 1&micro;l will be 100pmol primer. In the final qPCR reaction, the primers
      are used 1:400 times diluted to a final concentration of 250nM.
    </p>
    <p style="text-align: justify">
      Create a primer premix for the qPCR reaction using 2.5&micro;l forward primer stock solution, 2.5&micro;l 
      reverse primer stock solution and 95&micro;l water (diluting primer 1:40 each). In a 10&micro;l qPCR 
      reaction 1&micro;l of this stock is used (diluting 1:10).
    </p>
    <p style="text-align: justify">
      A typical qPCR reaction in 10&micro;l could be composed of 1&micro;l cDNA/genomic DNA (maximal 10ng), 
      1&micro;l primer premix, 3&micro;l water and 5&micro;l 2x qPCR mix (LightCycler&reg; 480 SYBR&reg; 
      Green I Master; 04887352001, Roche Diagnostics GmbH, Mannheim, Germany).
    </p>
    <p style="text-align: justify">
      Some further thoughts: In a 10&micro;l qPCR reaction will be 2.5pmol of each primer (equals 
      1.5*10<sup>12</sup> molecules each). If each primer pair would form an amplicon of 259bp length,
      the weight of the amplicons would be 427ng. In an experiment using FSTL1 279ng could be purified 
      after a regular qPCR reaction. This would suggest that 2 out of 3 primers (66%) formed a amplicon.
    </p>

    <h5 id="BASE-Genomic-Std">5.3. Use genomic DNA as Standard</h5>
    <p style="text-align: justify">
      Genomic DNA in combination with a single copy gene can be used as a standard. Use 4ng human genomic DNA 
      (Human Genomic DNA; 11691112001, Roche Diagnostics GmbH, Mannheim, Germany) per qPCR reaction containing 
      1200 copies of the genome and amplify the exon 6 of FSTL1 (follistatin like 1) in 6 technical replicates:
    </p>
<pre>
>FSTL1-fw
GGGAAAGGGTTTAAGTCCCC
>FSTL1-rv
TGGAGTAATGATGGAACAGGG
</pre>
    <p style="text-align: justify">
      Amplicon sequence (259 bp):<br />
      GGGAAAGGGTTTAAGTCCCCTCTCTCAGTTAGTCTTCCCTTGTGGATCATTCTCTGACCTCTCGGTTCCTCTCACCTAAGCAACTTGT
      TGTTCTCCTGGTCTGGATACGTTGTAATATTGATGGCAGTTTCATTCTGTTCCACAAACTTCAGGAATTCACTGGAGTCCAGGCGAGA
      ATCACCATTATCAAAGTTCTAGAAAGGGCATGACAAGATCGTTCAGGGATGTTTTGCAGAACCCCTGTTCCATCATTACTCCA
    </p>
    <p style="text-align: justify">
      Use RDML-Edit to provide the amplicon sequence in the target information to allow RDML-ExperimentAnalysis to compensate
      for the amplicon length. Without compensation (and in RDML-RunAnalysis) the Ncopy will be 3108 copies instead
      of the expected 1200 copies.
    </p>

  </main>

  <footer>
    <div class="container">
      <div class="float-left">
        <a class="plain" target="_blank" href="https://www.gear-genomics.com">
          GEAR + RDML Consortium &#126;
          <i class="fas fa-home"></i>
        </a>
        ·
        <a class="plain" target="_blank" href="https://github.com/gear-genomics">
          <i class="fab fa-github-alt"></i>
        </a>
        ·
        <a class="plain" target="_blank" href="http://rdml.org">
          <i class="fas fa-landmark" aria-hidden="true"></i>
          RDML Consortium 
        </a>
        ·
        <a class="plain" target="_blank" href="https://www.gear-genomics.com/terms">
          <i class="fa-solid fa-section" aria-hidden="true"></i>
          Terms of Use
        </a>
        ·
        <a class="plain" target="_blank" href="https://www.gear-genomics.com/contact">
          <i class="fa fa-envelope" aria-hidden="true"></i>
          Contact Us
        </a>
      </div>
      <div class="float-right">
        Supported by
        <a class="plain" target="_blank" href="https://www.embl.de/">
          <img class="footer-logo" alt="EMBL logo" src="static/img/embl.svg">
        </a>
      </div>
    </div>
  </footer>

  <script type="module" src="static/libs/bootstrap/4.0.0/js/bootstrap.min.js" crossorigin="anonymous"></script>

</body>

</html>
